<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.302">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Martin Proks">
<meta name="dcterms.date" content="2024-04-08">

<title>proks-salehin-et-al-2023 - 01 - NGS collection [mouse]</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script>
<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../notebooks/00_figures_common.html">Reports</a></li><li class="breadcrumb-item"><a href="../notebooks/01_fetchngs_mouse.html">01 - NGS collection [mouse]</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">proks-salehin-et-al-2023</a> 
        <div class="sidebar-tools-main">
    <a href="https://twitter.com/LabBrickman" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-twitter"></i></a>
    <a href="https://github.com/brickmanlab" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../README.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">proks-salehin-et-al</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Reports</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/00_figures_common.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">00 - Figures (common)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/00_figures_human.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">00 - Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/00_figures_mouse.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">00 - Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/01_fetchngs_human.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">01 - NGS collection [human]</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/01_fetchngs_human-invitro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">01 - NGS collection [human]</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/01_fetchngs_mouse.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">01 - NGS collection [mouse]</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/02_human_integration.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">02 - human integration</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/02_human_tunning.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">02 - human tunning</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/02_mouse_integration.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">02 - mouse integration</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/02_mouse_tunning.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">02 - mouse tunning</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/03_human_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">03 - human analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/03_mouse_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">03 - mouse analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/04_human_cellrank.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Human CellRank</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/04_mouse_cellrank.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">04 - CellRank</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/05_human_classifier.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Re-annotation and classification</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/05_mouse_classifier.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">05 - mouse classifier</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/05_mouse_classifier_accuracy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">05 - mouse classifier accuracy</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/06_human_query_Rivron_SRR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Rivron blastoids cleaning</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/06_mouse_query.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">06 - mouse query</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/08_shap_explainers.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SHAP explainers for SCANVI</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/09_classification_explanation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">09 - Classification explanation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/10_mouse_beyond_celltypes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">05 - mouse classifier</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/11_human_query_Smith.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Austin Smith paper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/11_xgboost_vs_scanvi.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11 - XGBoost vs scANVI</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/12_upload_models.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Upload AI models</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#datasets" id="toc-datasets" class="nav-link active" data-scroll-target="#datasets"><span class="header-section-number">1</span> Datasets</a>
  <ul class="collapse">
  <li><a href="#deng-et-al.-2014-gse45719" id="toc-deng-et-al.-2014-gse45719" class="nav-link" data-scroll-target="#deng-et-al.-2014-gse45719"><span class="header-section-number">1.1</span> Deng et al., 2014 GSE45719</a></li>
  <li><a href="#biase-et-al.-2014-gse57249" id="toc-biase-et-al.-2014-gse57249" class="nav-link" data-scroll-target="#biase-et-al.-2014-gse57249"><span class="header-section-number">1.2</span> Biase et al., 2014 GSE57249</a></li>
  <li><a href="#posfai-et-al.-2017-gse84892" id="toc-posfai-et-al.-2017-gse84892" class="nav-link" data-scroll-target="#posfai-et-al.-2017-gse84892"><span class="header-section-number">1.3</span> Posfai et al., 2017 GSE84892</a></li>
  <li><a href="#goolamn-et-al.-2016-e-mtab-3321" id="toc-goolamn-et-al.-2016-e-mtab-3321" class="nav-link" data-scroll-target="#goolamn-et-al.-2016-e-mtab-3321"><span class="header-section-number">1.4</span> Goolamn et al., 2016 E-MTAB-3321</a></li>
  <li><a href="#boroviak-et-al.-2015-e-mtab-2958-and-e-mtab-2959" id="toc-boroviak-et-al.-2015-e-mtab-2958-and-e-mtab-2959" class="nav-link" data-scroll-target="#boroviak-et-al.-2015-e-mtab-2958-and-e-mtab-2959"><span class="header-section-number">1.5</span> Boroviak et al., 2015 [E-MTAB-2958] and [E-MTAB-2959]</a></li>
  <li><a href="#chen-et-al.-2016-gse74155" id="toc-chen-et-al.-2016-gse74155" class="nav-link" data-scroll-target="#chen-et-al.-2016-gse74155"><span class="header-section-number">1.6</span> Chen et al., 2016 GSE74155</a></li>
  <li><a href="#nowotschin-et-al.-2019-gse123046" id="toc-nowotschin-et-al.-2019-gse123046" class="nav-link" data-scroll-target="#nowotschin-et-al.-2019-gse123046"><span class="header-section-number">1.7</span> Nowotschin et al., 2019 [GSE123046]</a></li>
  <li><a href="#fan-et-al.-2015-gse53386" id="toc-fan-et-al.-2015-gse53386" class="nav-link" data-scroll-target="#fan-et-al.-2015-gse53386"><span class="header-section-number">1.8</span> Fan et al., 2015 GSE53386</a></li>
  <li><a href="#borensztein-et-al.-2017-gse80810" id="toc-borensztein-et-al.-2017-gse80810" class="nav-link" data-scroll-target="#borensztein-et-al.-2017-gse80810"><span class="header-section-number">1.9</span> Borensztein et al., 2017 GSE80810</a></li>
  <li><a href="#stirparo-et-al.-2020-gse159030" id="toc-stirparo-et-al.-2020-gse159030" class="nav-link" data-scroll-target="#stirparo-et-al.-2020-gse159030"><span class="header-section-number">1.10</span> Stirparo et al., 2020 GSE159030</a></li>
  <li><a href="#xue-et-al.-2013-gse44183" id="toc-xue-et-al.-2013-gse44183" class="nav-link" data-scroll-target="#xue-et-al.-2013-gse44183"><span class="header-section-number">1.11</span> Xue et al., 2013 GSE44183</a></li>
  <li><a href="#yanagida-et-al.-2022-gse148462" id="toc-yanagida-et-al.-2022-gse148462" class="nav-link" data-scroll-target="#yanagida-et-al.-2022-gse148462"><span class="header-section-number">1.12</span> Yanagida et al., 2022 GSE148462</a></li>
  </ul></li>
  <li><a href="#mohammed-et-al.-2017-gse100597" id="toc-mohammed-et-al.-2017-gse100597" class="nav-link" data-scroll-target="#mohammed-et-al.-2017-gse100597"><span class="header-section-number">2</span> Mohammed et al., 2017 GSE100597</a>
  <ul class="collapse">
  <li><a href="#preparation-12" id="toc-preparation-12" class="nav-link" data-scroll-target="#preparation-12"><span class="header-section-number">2.1</span> Preparation</a></li>
  <li><a href="#load-data" id="toc-load-data" class="nav-link" data-scroll-target="#load-data"><span class="header-section-number">2.2</span> Load data</a></li>
  </ul></li>
  <li><a href="#integration" id="toc-integration" class="nav-link" data-scroll-target="#integration"><span class="header-section-number">3</span> Integration</a></li>
  <li><a href="#final-adjustments" id="toc-final-adjustments" class="nav-link" data-scroll-target="#final-adjustments"><span class="header-section-number">4</span> Final adjustments</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">01 - NGS collection [mouse]</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Martin Proks </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">April 8, 2024</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://rep.bioscientifica.com/view/journals/rep/145/3/images/large/R65fig1.jpeg" class="figure-img"></p>
<p></p><figcaption class="figure-caption">Timepoints were adjusted based on <a href="https://rep.bioscientifica.com/view/journals/rep/145/3/R65.xml">Saiz and Plusa et al., Reproduction 2013</a></figcaption><p></p>
</figure>
</div>
<p><strong>Mouse</strong>: - <del><a href="https://www.sciencedirect.com/science/article/pii/S1534580710001103">Guo et al., 2010</a>: qPCR on steroids, complicated, skipping - maybe for prediction, could be interesting)</del> - <del><a href="https://www.nature.com/articles/nature13544">Guo et al., 2014</a>: Bisulfite-Seq (skipping)</del> - ~~<a href="https://www.nature.com/articles/ncb2881">Ohnishi et al., 2014</a>: Microarray (used in ccmnetplus for validation) *~~</p>
<div class="cell" data-execution_count="1">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="op">!</span>which pip</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>/projects/dan1/data/Brickman/conda/envs/scvi-1.0.0/bin/pip</code></pre>
</div>
</div>
<div class="cell" data-execution_count="2">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> anndata</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> scanpy <span class="im">as</span> sc</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> load_experiment(filename: <span class="bu">str</span>, GEO: <span class="bu">str</span>, left_on: <span class="bu">str</span> <span class="op">=</span> <span class="st">'sample'</span>, right_on<span class="op">=</span><span class="st">'SRX'</span>):</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>    adata <span class="op">=</span> sc.read(filename)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>    metadata <span class="op">=</span> pd.read_csv(<span class="ss">f"../pipeline/fetchngs/</span><span class="sc">{</span>GEO<span class="sc">}</span><span class="ss">_metadata.csv"</span>)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>    <span class="co"># sometimes we have duplicates because the ID is not unique</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>    adata.obs <span class="op">=</span> pd.merge(adata.obs, metadata, left_on<span class="op">=</span>left_on, right_on<span class="op">=</span>right_on)<span class="op">\</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>                    .drop_duplicates(left_on)<span class="op">\</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>                    .set_index(left_on)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>    adata.obs[<span class="st">'ct_orig'</span>] <span class="op">=</span> adata.obs[<span class="st">'sample_title'</span>]</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> adata</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="3">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>run ..<span class="op">/</span>scripts<span class="op">/</span>helpers.py</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>/projects/dan1/data/Brickman/conda/envs/scvi-1.0.0/lib/python3.10/site-packages/scvi/_settings.py:63: UserWarning: Since v1.0.0, scvi-tools no longer uses a random seed by default. Run `scvi.settings.seed = 0` to reproduce results from previous versions.
  self.seed = seed
/projects/dan1/data/Brickman/conda/envs/scvi-1.0.0/lib/python3.10/site-packages/scvi/_settings.py:70: UserWarning: Setting `dl_pin_memory_gpu_training` is deprecated in v1.0 and will be removed in v1.1. Please pass in `pin_memory` to the data loaders instead.
  self.dl_pin_memory_gpu_training = (
/projects/dan1/data/Brickman/conda/envs/scvi-1.0.0/lib/python3.10/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm</code></pre>
</div>
</div>
<div class="cell" data-execution_count="4">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> normalize_smartseq(adata: sc.AnnData, gene_len: pd.DataFrame) <span class="op">-&gt;</span> sc.AnnData:</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"SMART-SEQ: Normalization"</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>    common_genes <span class="op">=</span> adata.var_names.intersection(gene_len.index)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"SMART-SEQ: Common genes </span><span class="sc">{</span>common_genes<span class="sc">.</span>shape[<span class="dv">0</span>]<span class="sc">}</span><span class="ss">"</span>)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>    lengths <span class="op">=</span> gene_len.loc[common_genes, <span class="st">"length"</span>].values</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>    normalized <span class="op">=</span> adata[:, common_genes].copy()</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>    normalized.X <span class="op">=</span> normalized.X <span class="op">/</span> lengths <span class="op">*</span> np.median(lengths)</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>    normalized.X <span class="op">=</span> np.rint(normalized.X)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> normalized</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="5">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># pip install gtftools</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co"># !gtftools \</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     -l ../data/external/Mus_musculus_GRCm38_102_gene_length.txt \</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co">#     /scratch/Brickman/references/mus_musculus/ensembl/GRCm38_102/Mus_musculus.GRCm38.102.gtf</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>genes_length <span class="op">=</span> pd.read_table(<span class="st">"../data/external/Mus_musculus_GRCm38_102_gene_length.txt"</span>).set_index(<span class="st">'gene'</span>)</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>genes_length[<span class="st">'length'</span>] <span class="op">=</span> genes_length[<span class="st">'mean'</span>]</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>genes_length</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="5">
<div>


<table class="dataframe table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">mean</th>
<th data-quarto-table-cell-role="th">median</th>
<th data-quarto-table-cell-role="th">longest_isoform</th>
<th data-quarto-table-cell-role="th">merged</th>
<th data-quarto-table-cell-role="th">length</th>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">gene</th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">ENSMUSG00000102693</td>
<td>1070</td>
<td>1070</td>
<td>1070</td>
<td>1070</td>
<td>1070</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">ENSMUSG00000064842</td>
<td>110</td>
<td>110</td>
<td>110</td>
<td>110</td>
<td>110</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">ENSMUSG00000051951</td>
<td>3592</td>
<td>3634</td>
<td>4153</td>
<td>6094</td>
<td>3592</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">ENSMUSG00000102851</td>
<td>480</td>
<td>480</td>
<td>480</td>
<td>480</td>
<td>480</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">ENSMUSG00000103377</td>
<td>2819</td>
<td>2819</td>
<td>2819</td>
<td>2819</td>
<td>2819</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">ENSMUSG00000117740</td>
<td>331</td>
<td>331</td>
<td>331</td>
<td>331</td>
<td>331</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">ENSMUSG00000117782</td>
<td>1570</td>
<td>1570</td>
<td>1570</td>
<td>1570</td>
<td>1570</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">ENSMUSG00000117951</td>
<td>256</td>
<td>256</td>
<td>256</td>
<td>256</td>
<td>256</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">ENSMUSG00000095993</td>
<td>168</td>
<td>168</td>
<td>168</td>
<td>168</td>
<td>168</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">ENSMUSG00000118197</td>
<td>243</td>
<td>243</td>
<td>243</td>
<td>243</td>
<td>243</td>
</tr>
</tbody>
</table>

<p>55364 rows × 5 columns</p>
</div>
</div>
</div>
<section id="datasets" class="level2" data-number="1">
<h2 data-number="1" data-anchor-id="datasets"><span class="header-section-number">1</span> Datasets</h2>
<table class="table">
<thead>
<tr class="header">
<th>Dataset</th>
<th>Technology</th>
<th>Download</th>
<th>Notes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><a href="https://www.science.org/doi/10.1126/science.1245316">Deng et al., 2014</a></td>
<td>SMART-seq 1/2</td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE45719">GSE45719</a></td>
<td>Done</td>
</tr>
<tr class="even">
<td><a href="https://genome.cshlp.org/content/24/11/1787">Biase et al., 2014</a></td>
<td>SMART-seq</td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE57249">GSE57249</a></td>
<td>Done</td>
</tr>
<tr class="odd">
<td><a href="https://elifesciences.org/articles/22906">Posfai et al., 2017</a></td>
<td>SMART-seq</td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE84892">GSE84892</a></td>
<td>Done</td>
</tr>
<tr class="even">
<td><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4819611/">Goolamn et al., 2016</a></td>
<td>SMART-seq2</td>
<td><a href="https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-3321/">E-MTAB-3321</a></td>
<td>Done</td>
</tr>
<tr class="odd">
<td><a href="https://www.sciencedirect.com/science/article/pii/S1534580715006589">Boroviak et al., 2015</a></td>
<td>SMART-seq2</td>
<td><a href="https://www.sciencedirect.com/science/article/pii/S1534580715006589">Sup Table S1</a></td>
<td>Done</td>
</tr>
<tr class="even">
<td><a href="https://genome.cshlp.org/content/26/10/1342.long">Chen et al., 2016</a></td>
<td>SMART-seq2</td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE74155">GSE74155</a></td>
<td>Done</td>
</tr>
<tr class="odd">
<td><a href="https://www.nature.com/articles/s41586-019-1127-1">Nowotschin et al., 2019</a></td>
<td>10X v2</td>
<td><a href="https://s3.amazonaws.com/dp-lab-data-public/mouse_endoderm/sc_endoderm_all_cells.h5ad">Portal</a></td>
<td>Done</td>
</tr>
<tr class="even">
<td><a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0706-1">Fan et al., 2015</a></td>
<td>SUPeR-seq</td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE53386">GSE53386</a></td>
<td>Done</td>
</tr>
<tr class="odd">
<td><a href="https://www.nature.com/articles/nsmb.3365">Borensztein et al., 2017</a></td>
<td>qRT-PCR</td>
<td><a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE80810">GSE80810</a></td>
<td>Done</td>
</tr>
<tr class="even">
<td><a href="https://www.pnas.org/doi/full/10.1073/pnas.2008890118">Stirparo et al., 2020</a></td>
<td>SMART-seq2</td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE159030">GSE159030</a></td>
<td>Done</td>
</tr>
<tr class="odd">
<td><a href="https://www.nature.com/articles/nature12364">Xue et al., 2013</a></td>
<td>qRT-PCR</td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE44183">GSE44183</a></td>
<td>Done</td>
</tr>
<tr class="even">
<td><a href="https://www.sciencedirect.com/science/article/pii/S0092867422001234">Yanagida et al., 2022</a></td>
<td>SMART-seq2</td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE148462">GSE148462</a></td>
<td>Done</td>
</tr>
<tr class="odd">
<td><a href="https://www.sciencedirect.com/science/article/pii/S2211124717309610">Mohammed et al., 2017</a></td>
<td>SMART-seq2</td>
<td><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE100597">GSE100597</a></td>
<td>Done</td>
</tr>
</tbody>
</table>
<section id="deng-et-al.-2014-gse45719" class="level3" data-number="1.1">
<h3 data-number="1.1" data-anchor-id="deng-et-al.-2014-gse45719"><span class="header-section-number">1.1</span> Deng et al., 2014 <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE45719">GSE45719</a></h3>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> <span class="dt">\</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>    Deng_et_al_2014 <span class="dt">\</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>    nextflow run nf-core/fetchngs <span class="dt">\</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> 1.10.0 <span class="dt">\</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/fetchngs/GSE45719.txt</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> Deng_et_al_ALIGNED nextflow run brickmanlab/scrnaseq <span class="dt">\</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> feature/smartseq <span class="dt">\</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">-c</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/smartseq.config <span class="dt">\</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">--protocol</span> smartseq <span class="dt">\</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">--aligner</span> star <span class="dt">\</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">--fasta</span> /scratch/Brickman/references/mus_musculus/ensembl/GRCm38_102/Mus_musculus.GRCm38.dna_sm.primary_assembly.fa <span class="dt">\</span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">--gtf</span> /scratch/Brickman/references/mus_musculus/ensembl/GRCm38_102/Mus_musculus.GRCm38.102.gtf <span class="dt">\</span></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">--star_index</span> /scratch/Brickman/references/mus_musculus/ensembl/GRCm38_102/star_2.7.10a <span class="dt">\</span></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /scratch/Brickman/pipelines/Deng_et_al_2014/results/samplesheet/samplesheet.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="preparation" class="level4" data-number="1.1.1">
<h4 data-number="1.1.1" data-anchor-id="preparation"><span class="header-section-number">1.1.1</span> Preparation</h4>
<div class="cell" data-execution_count="6">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>deng_metadata <span class="op">=</span> pd.read_table(<span class="st">"https://ftp.ncbi.nlm.nih.gov/geo/series/GSE45nnn/GSE45719/matrix/GSE45719_series_matrix.txt.gz"</span>, </span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>                              skiprows<span class="op">=</span><span class="dv">30</span>, index_col<span class="op">=</span><span class="dv">0</span>).T</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="7">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>deng_metadata[<span class="st">'ct'</span>] <span class="op">=</span> deng_metadata[<span class="st">'!Sample_source_name_ch1'</span>].values</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>deng_metadata[<span class="st">'SRX'</span>] <span class="op">=</span> deng_metadata[[<span class="st">'!Sample_relation'</span>]].agg(<span class="st">' '</span>.join, axis<span class="op">=</span><span class="dv">1</span>).<span class="bu">str</span>.extract(<span class="vs">r'(SRX[0-9]</span><span class="sc">{6}</span><span class="vs">)'</span>)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>deng_metadata <span class="op">=</span> deng_metadata.reset_index()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="8">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>legend <span class="op">=</span> {</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">'16-cell stage blastomere'</span>: <span class="st">'16C'</span>, </span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">'2-cell stage blastomere'</span>: <span class="st">'2C'</span>,</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">'4-cell stage blastomere'</span>: <span class="st">'4C'</span>,</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">'8-cell stage blastomere'</span>: <span class="st">'8C'</span>,</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Early 2-cell stage blastomere (31-32h post-fertilization)'</span>: <span class="st">'2C'</span>,</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Early blastocyst cell (86-88h post-fertilization)'</span>: <span class="st">'Blastocyst'</span>,</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Late 2-cell stage blastomere (46-48h post-fertilization)'</span>: <span class="st">'2C'</span>,</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Late blastocyst cell (100-102h post-fertilization)'</span>: <span class="st">'Blastocyst'</span>,</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Liver'</span>: <span class="st">'Liver'</span>,</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Liver cell'</span>: <span class="st">'Liver'</span>, </span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Mid 2-cell stage blastomere (34-40h post-fertilization)'</span>: <span class="st">'2C'</span>,</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Mid blastocyst cell (92-94h post-fertilization)'</span>: <span class="st">'Blastocyst'</span>,</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Zygote'</span>: <span class="st">'Zygote'</span>,</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">'fibroblast (primary culture from tail)'</span>: <span class="st">'Fibroblast'</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> old_name, new_name <span class="kw">in</span> legend.items():</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>    deng_metadata.loc[deng_metadata.ct <span class="op">==</span> old_name, <span class="st">'ct'</span>] <span class="op">=</span> new_name</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>deng_metadata.ct <span class="op">=</span> deng_metadata.ct.astype(<span class="st">'category'</span>)</span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>deng_metadata <span class="op">=</span> deng_metadata.query(<span class="st">'ct not in ["Liver", "Blastocyst", "Fibroblast"]'</span>).copy()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="9">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>deng_metadata[<span class="st">'index'</span>].to_csv(<span class="st">"../pipeline/fetchngs/GSE45719.txt"</span>, index<span class="op">=</span><span class="va">None</span>, header<span class="op">=</span><span class="va">None</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>deng_metadata.set_index(<span class="st">'SRX'</span>).to_csv(<span class="st">"../pipeline/fetchngs/GSE45719_metadata.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-dataset" class="level4" data-number="1.1.2">
<h4 data-number="1.1.2" data-anchor-id="load-dataset"><span class="header-section-number">1.1.2</span> Load dataset</h4>
<div class="cell" data-execution_count="10">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>deng <span class="op">=</span> load_experiment(</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../data/external/aligned/mouse/Deng_et_al_ALIGNED/results/star/mtx_conversions/combined_matrix.h5ad"</span>,</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>    GEO<span class="op">=</span><span class="st">"GSE45719"</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="11">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>deng.obs[<span class="st">'experiment'</span>] <span class="op">=</span> <span class="st">"Deng et al., 2014"</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>deng.obs[<span class="st">'technology'</span>] <span class="op">=</span> <span class="st">'SMART-seq'</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>deng.obs.loc[deng.obs.sample_alias.isin([<span class="ss">f'GSM12780</span><span class="sc">{</span>i<span class="sc">:02d}</span><span class="ss">'</span> <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="dv">9</span>, <span class="dv">45</span>)]), <span class="st">'technology'</span>] <span class="op">=</span> <span class="st">'SMART-seq2'</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>deng.obs[<span class="st">'batch'</span>] <span class="op">=</span> <span class="st">"DENG_1"</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>deng.obs.loc[deng.obs.technology <span class="op">==</span> <span class="st">'SMART-seq2'</span>, <span class="st">'batch'</span>] <span class="op">=</span> <span class="st">"DENG_2"</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>deng.obs <span class="op">=</span> deng.obs[[<span class="st">'batch'</span>, <span class="st">'experiment'</span>, <span class="st">'technology'</span>, <span class="st">'ct'</span>, <span class="st">'ct_orig'</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="12">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>deng <span class="op">=</span> normalize_smartseq(deng, genes_length)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>SMART-SEQ: Normalization
SMART-SEQ: Common genes 55364</code></pre>
</div>
</div>
</section>
</section>
<section id="biase-et-al.-2014-gse57249" class="level3" data-number="1.2">
<h3 data-number="1.2" data-anchor-id="biase-et-al.-2014-gse57249"><span class="header-section-number">1.2</span> Biase et al., 2014 <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE57249">GSE57249</a></h3>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> <span class="dt">\</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>    Biase_et_al <span class="dt">\</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>    nextflow run nf-core/fetchngs <span class="dt">\</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> 1.10.0 <span class="dt">\</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/fetchngs/GSE57249.txt</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> Biase_et_al nextflow run brickmanlab/scrnaseq <span class="dt">\</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> feature/smartseq <span class="dt">\</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">-c</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/smartseq.config <span class="dt">\</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /scratch/Brickman/pipelines/Biase_et_al/results/samplesheet/samplesheet.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="preparation-1" class="level4" data-number="1.2.1">
<h4 data-number="1.2.1" data-anchor-id="preparation-1"><span class="header-section-number">1.2.1</span> Preparation</h4>
<div class="cell" data-execution_count="13">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>biase_metadata <span class="op">=</span> pd.concat([</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>    pd.read_table(<span class="st">"https://ftp.ncbi.nlm.nih.gov/geo/series/GSE57nnn/GSE57249/matrix/GSE57249-GPL13112_series_matrix.txt.gz"</span>,</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>                  skiprows<span class="op">=</span><span class="dv">33</span>, index_col<span class="op">=</span><span class="dv">0</span>).T.reset_index().set_index(<span class="st">'!Sample_geo_accession'</span>),</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>    pd.read_table(<span class="st">"https://ftp.ncbi.nlm.nih.gov/geo/series/GSE57nnn/GSE57249/matrix/GSE57249-GPL17021_series_matrix.txt.gz"</span>,</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>                  skiprows<span class="op">=</span><span class="dv">33</span>, index_col<span class="op">=</span><span class="dv">0</span>).T.reset_index().set_index(<span class="st">'!Sample_geo_accession'</span>)</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="14">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>biase_metadata[<span class="st">'ct'</span>] <span class="op">=</span> biase_metadata[<span class="st">'index'</span>].values</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>biase_metadata[<span class="st">'SRX'</span>] <span class="op">=</span> biase_metadata[[<span class="st">'!Sample_relation'</span>]].agg(<span class="st">' '</span>.join, axis<span class="op">=</span><span class="dv">1</span>).<span class="bu">str</span>.extract(<span class="vs">r'(SRX[0-9]</span><span class="sc">{6}</span><span class="vs">)'</span>)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>biase_metadata <span class="op">=</span> biase_metadata.reset_index()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="15">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>legend <span class="op">=</span> {</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Zygote'</span>: <span class="st">'Zygote'</span>,</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Two-cell'</span>: <span class="st">'2C'</span>,</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Four-cell'</span>: <span class="st">'4C'</span>,</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Inner cell mass'</span>: <span class="st">'E3.5-ICM'</span>,</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Trophectoderm'</span>: <span class="st">'E3.5-TE'</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> old_name, new_name <span class="kw">in</span> legend.items():</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>    biase_metadata.loc[biase_metadata.ct.<span class="bu">str</span>.startswith(old_name), <span class="st">'ct'</span>] <span class="op">=</span> new_name</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>biase_metadata.ct <span class="op">=</span> biase_metadata.ct.astype(<span class="st">'category'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="16">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>biase_metadata[<span class="st">'!Sample_geo_accession'</span>].to_csv(<span class="st">"../pipeline/fetchngs/GSE57249.txt"</span>, index<span class="op">=</span><span class="va">None</span>, header<span class="op">=</span><span class="va">None</span>)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>biase_metadata.set_index(<span class="st">'SRX'</span>).to_csv(<span class="st">"../pipeline/fetchngs/GSE57249_metadata.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-dataset-1" class="level4" data-number="1.2.2">
<h4 data-number="1.2.2" data-anchor-id="load-dataset-1"><span class="header-section-number">1.2.2</span> Load dataset</h4>
<div class="cell" data-execution_count="17">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>biase <span class="op">=</span> load_experiment(</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../data/external/aligned/mouse/Biase_et_al/results/star/mtx_conversions/combined_matrix.h5ad"</span>,</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>    GEO<span class="op">=</span><span class="st">"GSE57249"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="18">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>biase.obs[<span class="st">'batch'</span>] <span class="op">=</span> <span class="st">"BIASE_1"</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>biase.obs[<span class="st">'experiment'</span>] <span class="op">=</span> <span class="st">"Biase et al., 2014"</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>biase.obs[<span class="st">'technology'</span>] <span class="op">=</span> <span class="st">"SMART-seq2"</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>biase.obs <span class="op">=</span> biase.obs[[<span class="st">'batch'</span>, <span class="st">'experiment'</span>, <span class="st">'technology'</span>, <span class="st">'ct'</span>, <span class="st">'ct_orig'</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="19">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>biase <span class="op">=</span> normalize_smartseq(biase, genes_length)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>SMART-SEQ: Normalization
SMART-SEQ: Common genes 55364</code></pre>
</div>
</div>
<div class="cell" data-execution_count="20">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>biase</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="20">
<pre><code>AnnData object with n_obs × n_vars = 56 × 55364
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig'
    var: 'gene_symbol'</code></pre>
</div>
</div>
</section>
</section>
<section id="posfai-et-al.-2017-gse84892" class="level3" data-number="1.3">
<h3 data-number="1.3" data-anchor-id="posfai-et-al.-2017-gse84892"><span class="header-section-number">1.3</span> Posfai et al., 2017 <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE84892">GSE84892</a></h3>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> <span class="dt">\</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>    Posfai_et_al <span class="dt">\</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>    nextflow run nf-core/fetchngs <span class="dt">\</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> 1.10.0 <span class="dt">\</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/fetchngs/GSE84892.txt</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> Posfai_et_al nextflow run brickmanlab/scrnaseq <span class="dt">\</span></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> feature/smartseq <span class="dt">\</span></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">-c</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/smartseq.config <span class="dt">\</span></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /scratch/Brickman/pipelines/Posfai_et_al/results/samplesheet/samplesheet.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="preparation-2" class="level4" data-number="1.3.1">
<h4 data-number="1.3.1" data-anchor-id="preparation-2"><span class="header-section-number">1.3.1</span> Preparation</h4>
<div class="cell" data-execution_count="21">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>posfai_metadata <span class="op">=</span> pd.read_table(<span class="st">"https://ftp.ncbi.nlm.nih.gov/geo/series/GSE84nnn/GSE84892/matrix/GSE84892_series_matrix.txt.gz"</span>,</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>                                skiprows<span class="op">=</span><span class="dv">31</span>, index_col<span class="op">=</span><span class="dv">0</span>).T</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="22">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>posfai_metadata[<span class="st">'SRX'</span>] <span class="op">=</span> posfai_metadata[[<span class="st">'!Sample_relation'</span>]].agg(<span class="st">' '</span>.join, axis<span class="op">=</span><span class="dv">1</span>).<span class="bu">str</span>.extract(<span class="vs">r'(SRX[0-9]</span><span class="sc">{6}</span><span class="vs">)'</span>)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>posfai_metadata <span class="op">=</span> posfai_metadata.reset_index()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="23">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>legend <span class="op">=</span> {</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">'16'</span>: <span class="st">'16C-'</span>,</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">'32'</span>: <span class="st">'E3.25-'</span>,</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">'64'</span>: <span class="st">'E3.5-'</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="24">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>posfai_metadata[<span class="st">'ct'</span>] <span class="op">=</span> <span class="st">'UNKNOWN'</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> old_name, new_name <span class="kw">in</span> legend.items():</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>    posfai_metadata.loc[posfai_metadata.loc[:, <span class="st">'!Sample_characteristics_ch1'</span>].iloc[:, <span class="dv">0</span>].<span class="bu">str</span>.contains(old_name), <span class="st">'ct'</span>] <span class="op">=</span> new_name</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>posfai_metadata.ct <span class="op">=</span> posfai_metadata.ct <span class="op">+</span> posfai_metadata.loc[:, <span class="st">'!Sample_characteristics_ch1'</span>].iloc[:, <span class="dv">2</span>].<span class="bu">str</span>.replace(<span class="st">'lineage: '</span>, <span class="st">''</span>)</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>posfai_metadata <span class="op">=</span> posfai_metadata.query(<span class="st">'~ct.str.contains("CO") or ct != "UNKNOWN"'</span>).copy()</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>posfai_metadata.loc[posfai_metadata.ct.<span class="bu">str</span>.contains(<span class="st">'16C'</span>), <span class="st">'ct'</span>] <span class="op">=</span> <span class="st">'16C'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="25">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>posfai_metadata[<span class="st">'!Sample_geo_accession'</span>].to_csv(<span class="st">"../pipeline/fetchngs/GSE84892.txt"</span>, index<span class="op">=</span><span class="va">None</span>, header<span class="op">=</span><span class="va">None</span>)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>posfai_metadata.set_index(<span class="st">'SRX'</span>).to_csv(<span class="st">"../pipeline/fetchngs/GSE84892_metadata.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-dataset-2" class="level4" data-number="1.3.2">
<h4 data-number="1.3.2" data-anchor-id="load-dataset-2"><span class="header-section-number">1.3.2</span> Load dataset</h4>
<div class="cell" data-execution_count="26">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>posfai <span class="op">=</span> sc.read(<span class="st">"../data/external/aligned/mouse/Posfai_et_al/results/star/mtx_conversions/combined_matrix.h5ad"</span>)</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>posfai <span class="op">=</span> posfai[<span class="op">~</span>posfai.obs.duplicated()].copy()</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>posfai_metadata <span class="op">=</span> pd.read_csv(<span class="st">"../pipeline/fetchngs/GSE84892_metadata.csv"</span>)</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>posfai.obs <span class="op">=</span> pd.merge(posfai.obs, posfai_metadata, left_on<span class="op">=</span><span class="st">"sample_title"</span>, right_on<span class="op">=</span><span class="st">"index"</span>).set_index(<span class="st">'sample_title'</span>)</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>posfai <span class="op">=</span> posfai[posfai.obs.ct <span class="op">!=</span> <span class="st">"E3.25-CO"</span>].copy()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>/projects/dan1/data/Brickman/conda/envs/scvi-1.0.0/lib/python3.10/site-packages/anndata/_core/anndata.py:1830: UserWarning: Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
  utils.warn_names_duplicates("obs")</code></pre>
</div>
</div>
<div class="cell" data-execution_count="27">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>posfai.obs[<span class="st">'batch'</span>] <span class="op">=</span> <span class="st">"POSFAI_1"</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>posfai.obs[<span class="st">'experiment'</span>] <span class="op">=</span> <span class="st">"Posfai et al., 2017"</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>posfai.obs[<span class="st">'technology'</span>] <span class="op">=</span> <span class="st">"SMART-seq2"</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>posfai.obs[<span class="st">'ct_orig'</span>] <span class="op">=</span> posfai.obs[<span class="st">'!Sample_characteristics_ch1'</span>]</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>posfai.obs <span class="op">=</span> posfai.obs[[<span class="st">'batch'</span>, <span class="st">'experiment'</span>, <span class="st">'technology'</span>, <span class="st">'ct'</span>, <span class="st">'ct_orig'</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="28">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>posfai <span class="op">=</span> normalize_smartseq(posfai, genes_length)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>SMART-SEQ: Normalization
SMART-SEQ: Common genes 55364</code></pre>
</div>
</div>
<div class="cell" data-execution_count="29">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>posfai</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="29">
<pre><code>AnnData object with n_obs × n_vars = 261 × 55364
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig'
    var: 'gene_symbol'</code></pre>
</div>
</div>
</section>
</section>
<section id="goolamn-et-al.-2016-e-mtab-3321" class="level3" data-number="1.4">
<h3 data-number="1.4" data-anchor-id="goolamn-et-al.-2016-e-mtab-3321"><span class="header-section-number">1.4</span> Goolamn et al., 2016 <a href="https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-3321/">E-MTAB-3321</a></h3>
<div class="sourceCode" id="cb41"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> <span class="dt">\</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>    Goolam_et_al <span class="dt">\</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>    nextflow run nf-core/fetchngs <span class="dt">\</span></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> 1.10.0 <span class="dt">\</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/fetchngs/E-MTAB-3321.txt</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> Goolam_et_al nextflow run brickmanlab/scrnaseq <span class="dt">\</span></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> feature/smartseq <span class="dt">\</span></span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">-c</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/smartseq.config <span class="dt">\</span></span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /scratch/Brickman/pipelines/Goolam_et_al/results/samplesheet/samplesheet.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="preparation-3" class="level4" data-number="1.4.1">
<h4 data-number="1.4.1" data-anchor-id="preparation-3"><span class="header-section-number">1.4.1</span> Preparation</h4>
<div class="cell" data-execution_count="30">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>goolamn_metadata <span class="op">=</span> pd.read_table(<span class="st">"https://www.ebi.ac.uk/biostudies/files/E-MTAB-3321/E-MTAB-3321.sdrf.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="31">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>goolamn_metadata[<span class="st">'ct'</span>] <span class="op">=</span> goolamn_metadata[<span class="st">'Characteristics[developmental stage]'</span>].values</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="32">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>legend <span class="op">=</span> {</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">'cleavage 2-cell'</span> : <span class="st">'2C'</span>,</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">'cleavage 4-cell'</span>: <span class="st">'4C'</span>,</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">'cleavage 8-cell'</span>: <span class="st">'8C'</span>,</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">'cleavage 16-cell'</span>: <span class="st">'16C'</span>,</span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">'cleavage 32-cell'</span>: <span class="st">'32C'</span></span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> old_name, new_name <span class="kw">in</span> legend.items():</span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a>    goolamn_metadata.loc[goolamn_metadata.ct <span class="op">==</span> old_name, <span class="st">'ct'</span>] <span class="op">=</span> new_name</span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a>goolamn_metadata.ct <span class="op">=</span> goolamn_metadata.ct.astype(<span class="st">'category'</span>)</span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-13"><a href="#cb44-13" aria-hidden="true" tabindex="-1"></a>goolamn_metadata <span class="op">=</span> goolamn_metadata.query(<span class="st">'ct != "32C"'</span>).copy()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="33">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>goolamn_metadata[<span class="st">'Comment[ENA_RUN]'</span>].to_csv(<span class="st">"../pipeline/fetchngs/E-MTAB-3321.txt"</span>, index<span class="op">=</span><span class="va">None</span>, header<span class="op">=</span><span class="va">None</span>)</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>goolamn_metadata.set_index(<span class="st">'Comment[ENA_RUN]'</span>).to_csv(<span class="st">"../pipeline/fetchngs/E-MTAB-3321_metadata.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-dataset-3" class="level4" data-number="1.4.2">
<h4 data-number="1.4.2" data-anchor-id="load-dataset-3"><span class="header-section-number">1.4.2</span> Load dataset</h4>
<div class="cell" data-execution_count="34">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>goolam <span class="op">=</span> load_experiment(</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../data/external/aligned/mouse/Goolam_et_al/results/star/mtx_conversions/combined_matrix.h5ad"</span>,</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>    GEO<span class="op">=</span><span class="st">"E-MTAB-3321"</span>, left_on<span class="op">=</span><span class="st">'run_accession'</span>, right_on<span class="op">=</span><span class="st">'Comment[ENA_RUN]'</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="35">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>goolam.obs[<span class="st">'batch'</span>] <span class="op">=</span> [<span class="ss">f"GOOLAM_</span><span class="sc">{</span>batch<span class="sc">}</span><span class="ss">"</span> <span class="cf">for</span> batch <span class="kw">in</span> goolam.obs[<span class="st">"Characteristics[batch]"</span>].values]</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>goolam.obs[<span class="st">'experiment'</span>] <span class="op">=</span> <span class="st">"Goolamn et al., 2016"</span></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>goolam.obs[<span class="st">'technology'</span>] <span class="op">=</span> <span class="st">"SMART-seq2"</span></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>goolam.obs <span class="op">=</span> goolam.obs[[<span class="st">'batch'</span>, <span class="st">'experiment'</span>, <span class="st">'technology'</span>, <span class="st">'ct'</span>, <span class="st">'ct_orig'</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="36">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>goolam <span class="op">=</span> normalize_smartseq(goolam, genes_length)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>SMART-SEQ: Normalization
SMART-SEQ: Common genes 55364</code></pre>
</div>
</div>
<div class="cell" data-execution_count="37">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>goolam</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="37">
<pre><code>AnnData object with n_obs × n_vars = 118 × 55364
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig'
    var: 'gene_symbol'</code></pre>
</div>
</div>
</section>
</section>
<section id="boroviak-et-al.-2015-e-mtab-2958-and-e-mtab-2959" class="level3" data-number="1.5">
<h3 data-number="1.5" data-anchor-id="boroviak-et-al.-2015-e-mtab-2958-and-e-mtab-2959"><span class="header-section-number">1.5</span> Boroviak et al., 2015 [E-MTAB-2958] and [E-MTAB-2959]</h3>
<div class="sourceCode" id="cb52"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> <span class="dt">\</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>    Boroviak_et_al <span class="dt">\</span></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>    nextflow run nf-core/fetchngs <span class="dt">\</span></span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> 1.10.0 <span class="dt">\</span></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/fetchngs/E-MTAB-29589.txt</span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> Boroviak_et_al nextflow run brickmanlab/scrnaseq <span class="dt">\</span></span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> feature/smartseq <span class="dt">\</span></span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">-c</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/smartseq.config <span class="dt">\</span></span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /scratch/Brickman/pipelines/Boroviak_et_al/results/samplesheet/samplesheet.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="preparation-4" class="level4" data-number="1.5.1">
<h4 data-number="1.5.1" data-anchor-id="preparation-4"><span class="header-section-number">1.5.1</span> Preparation</h4>
<div class="cell" data-execution_count="38">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>boroviak_metadata <span class="op">=</span> pd.concat([</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>    pd.read_table(<span class="st">"https://ftp.ebi.ac.uk/biostudies/nfs/E-MTAB-/958/E-MTAB-2958/Files/E-MTAB-2958.sdrf.txt"</span>),</span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>    pd.read_table(<span class="st">"https://ftp.ebi.ac.uk/biostudies/nfs/E-MTAB-/959/E-MTAB-2959/Files/E-MTAB-2959.sdrf.txt"</span>)</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="39">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>boroviak_metadata[<span class="st">'ct'</span>] <span class="op">=</span> boroviak_metadata[<span class="st">'Extract Name'</span>].values</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="40">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>legend <span class="op">=</span> {</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">'E2.5_MOR'</span>: <span class="st">'8C'</span>,</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">'E3.5_ICM'</span>: <span class="st">'E3.5-ICM'</span>,</span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">'E4.5_PrE'</span>: <span class="st">'E4.5-PrE'</span>,</span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">'E4.5_EPI'</span>: <span class="st">'E4.5-EPI'</span>,</span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> old_name, new_name <span class="kw">in</span> legend.items():</span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a>    boroviak_metadata.loc[boroviak_metadata.ct.<span class="bu">str</span>.startswith(old_name), <span class="st">'ct'</span>] <span class="op">=</span> new_name</span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a>boroviak_metadata <span class="op">=</span> boroviak_metadata.query(<span class="st">'ct in ["8C", "E3.5-ICM", "E4.5-EPI", "E4.5-PrE"]'</span>).copy()</span>
<span id="cb55-12"><a href="#cb55-12" aria-hidden="true" tabindex="-1"></a>boroviak_metadata.ct <span class="op">=</span> boroviak_metadata.ct.astype(<span class="st">'category'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="41">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>boroviak_metadata[<span class="st">'Comment[ENA_RUN]'</span>].to_csv(<span class="st">"../pipeline/fetchngs/E-MTAB-29589.txt"</span>, index<span class="op">=</span><span class="va">None</span>, header<span class="op">=</span><span class="va">None</span>)</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>boroviak_metadata.set_index(<span class="st">'Comment[ENA_RUN]'</span>).to_csv(<span class="st">"../pipeline/fetchngs/E-MTAB-29589_metadata.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-dataset-4" class="level4" data-number="1.5.2">
<h4 data-number="1.5.2" data-anchor-id="load-dataset-4"><span class="header-section-number">1.5.2</span> Load dataset</h4>
<div class="cell" data-execution_count="42">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>boroviak <span class="op">=</span> load_experiment(</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../data/external/aligned/mouse/Boroviak_et_al/results/star/mtx_conversions/combined_matrix.h5ad"</span>,</span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>    GEO<span class="op">=</span><span class="st">"E-MTAB-29589"</span>, left_on<span class="op">=</span><span class="st">'run_accession'</span>, right_on<span class="op">=</span><span class="st">'Comment[ENA_RUN]'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="43">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>boroviak.obs[<span class="st">'batch'</span>] <span class="op">=</span> <span class="st">"BOROVIAK_1"</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>boroviak.obs[<span class="st">'experiment'</span>] <span class="op">=</span> <span class="st">"Boroviak et al., 2015"</span></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>boroviak.obs[<span class="st">'technology'</span>] <span class="op">=</span> <span class="st">"SMART-seq2"</span></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>boroviak.obs <span class="op">=</span> boroviak.obs[[<span class="st">'batch'</span>, <span class="st">'experiment'</span>, <span class="st">'technology'</span>, <span class="st">'ct'</span>, <span class="st">'ct_orig'</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="44">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>boroviak <span class="op">=</span> normalize_smartseq(boroviak, genes_length)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>SMART-SEQ: Normalization
SMART-SEQ: Common genes 55364</code></pre>
</div>
</div>
<div class="cell" data-execution_count="45">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>boroviak</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="45">
<pre><code>AnnData object with n_obs × n_vars = 12 × 55364
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig'
    var: 'gene_symbol'</code></pre>
</div>
</div>
</section>
</section>
<section id="chen-et-al.-2016-gse74155" class="level3" data-number="1.6">
<h3 data-number="1.6" data-anchor-id="chen-et-al.-2016-gse74155"><span class="header-section-number">1.6</span> Chen et al., 2016 <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE74155">GSE74155</a></h3>
<div class="sourceCode" id="cb63"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> <span class="dt">\</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>    Chen_et_al <span class="dt">\</span></span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>    nextflow run nf-core/fetchngs <span class="dt">\</span></span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> 1.10.0 <span class="dt">\</span></span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">--force_sratools_download</span> <span class="dt">\</span></span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/fetchngs/GSE74155.txt</span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> Chen_et_al nextflow run brickmanlab/scrnaseq <span class="dt">\</span></span>
<span id="cb63-9"><a href="#cb63-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> feature/smartseq <span class="dt">\</span></span>
<span id="cb63-10"><a href="#cb63-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">-c</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/smartseq.config <span class="dt">\</span></span>
<span id="cb63-11"><a href="#cb63-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /scratch/Brickman/pipelines/Chen_et_al/results/samplesheet/samplesheet.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="preparation-5" class="level4" data-number="1.6.1">
<h4 data-number="1.6.1" data-anchor-id="preparation-5"><span class="header-section-number">1.6.1</span> Preparation</h4>
<div class="cell" data-execution_count="46">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>chen_metadata <span class="op">=</span> pd.read_table(<span class="st">"https://ftp.ncbi.nlm.nih.gov/geo/series/GSE74nnn/GSE74155/matrix/GSE74155_series_matrix.txt.gz"</span>,</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>                              skiprows<span class="op">=</span><span class="dv">30</span>, index_col<span class="op">=</span><span class="dv">0</span>).T</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="47">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>chen_metadata[<span class="st">'ct'</span>] <span class="op">=</span> chen_metadata[<span class="st">'!Sample_characteristics_ch1'</span>].iloc[:, <span class="dv">1</span>].values</span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>chen_metadata[<span class="st">'SRX'</span>] <span class="op">=</span> chen_metadata[[<span class="st">'!Sample_relation'</span>]].agg(<span class="st">' '</span>.join, axis<span class="op">=</span><span class="dv">1</span>).<span class="bu">str</span>.extract(<span class="vs">r'(SRX[0-9]</span><span class="sc">{6}</span><span class="vs">)'</span>)</span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a>chen_metadata <span class="op">=</span> chen_metadata.reset_index()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="48">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>legend <span class="op">=</span> {</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">'cell type: E3.5 ICM'</span>: <span class="st">'E3.5-ICM'</span>,</span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">'cell type: E4.5 Epiblast'</span>: <span class="st">'E4.5-EPI'</span></span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> old_name, new_name <span class="kw">in</span> legend.items():</span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a>    chen_metadata.loc[chen_metadata.ct.<span class="bu">str</span>.startswith(old_name), <span class="st">'ct'</span>] <span class="op">=</span> new_name</span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a>chen_metadata <span class="op">=</span> chen_metadata.query(<span class="st">'ct in ["E3.5-ICM", "E4.5-EPI"]'</span>).copy()</span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a>chen_metadata.ct <span class="op">=</span> chen_metadata.ct.astype(<span class="st">'category'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="49">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>chen_metadata[<span class="st">'!Sample_geo_accession'</span>].to_csv(<span class="st">"../pipeline/fetchngs/GSE74155.txt"</span>, index<span class="op">=</span><span class="va">None</span>, header<span class="op">=</span><span class="va">None</span>)</span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>chen_metadata.set_index(<span class="st">'SRX'</span>).to_csv(<span class="st">"../pipeline/fetchngs/GSE74155_metadata.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-dataset-5" class="level4" data-number="1.6.2">
<h4 data-number="1.6.2" data-anchor-id="load-dataset-5"><span class="header-section-number">1.6.2</span> Load dataset</h4>
<div class="cell" data-execution_count="50">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>chen <span class="op">=</span> load_experiment(</span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../data/external/aligned/mouse/Chen_et_al/results/star/mtx_conversions/combined_matrix.h5ad"</span>,</span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>    GEO<span class="op">=</span><span class="st">"GSE74155"</span>, left_on<span class="op">=</span><span class="st">'sample_alias'</span>, right_on<span class="op">=</span><span class="st">'ID_REF'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="51">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>chen.obs[<span class="st">'batch'</span>] <span class="op">=</span> <span class="st">"CHEN_1"</span></span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>chen.obs[<span class="st">'experiment'</span>] <span class="op">=</span> <span class="st">"Chen et al., 2016"</span></span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>chen.obs[<span class="st">'technology'</span>] <span class="op">=</span> <span class="st">"SMART-seq2"</span></span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a>chen.obs <span class="op">=</span> chen.obs[[<span class="st">'batch'</span>, <span class="st">'experiment'</span>, <span class="st">'technology'</span>, <span class="st">'ct'</span>, <span class="st">'ct_orig'</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="52">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>chen <span class="op">=</span> normalize_smartseq(chen, genes_length)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>SMART-SEQ: Normalization
SMART-SEQ: Common genes 55364</code></pre>
</div>
</div>
<div class="cell" data-execution_count="53">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>chen</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="53">
<pre><code>AnnData object with n_obs × n_vars = 46 × 55364
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig'
    var: 'gene_symbol'</code></pre>
</div>
</div>
</section>
</section>
<section id="nowotschin-et-al.-2019-gse123046" class="level3" data-number="1.7">
<h3 data-number="1.7" data-anchor-id="nowotschin-et-al.-2019-gse123046"><span class="header-section-number">1.7</span> Nowotschin et al., 2019 [GSE123046]</h3>
<div class="sourceCode" id="cb74"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> <span class="dt">\</span></span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>    Nowotschin_et_al <span class="dt">\</span></span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>    nextflow run nf-core/fetchngs <span class="dt">\</span></span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> 1.10.0 <span class="dt">\</span></span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">--force_sratools_download</span> <span class="dt">\</span></span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/fetchngs/GSE123046.txt</span>
<span id="cb74-7"><a href="#cb74-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-8"><a href="#cb74-8" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://raw.githubusercontent.com/10XGenomics/cellranger/master/lib/python/cellranger/barcodes/737K-august-2016.txt ../pipeline/737K-august-2016.txt <span class="kw">&amp;&amp;</span> <span class="fu">gzip</span> ../pipeline/737K-august-2016.txt</span>
<span id="cb74-9"><a href="#cb74-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-10"><a href="#cb74-10" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> Nowotschin_et_al nextflow run brickmanlab/scrnaseq <span class="dt">\</span></span>
<span id="cb74-11"><a href="#cb74-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> feature/smartseq <span class="dt">\</span></span>
<span id="cb74-12"><a href="#cb74-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">-c</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/10X.config <span class="dt">\</span></span>
<span id="cb74-13"><a href="#cb74-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /scratch/Brickman/pipelines/Nowotschin_et_al/results/samplesheet/samplesheet.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="preparation-6" class="level4" data-number="1.7.1">
<h4 data-number="1.7.1" data-anchor-id="preparation-6"><span class="header-section-number">1.7.1</span> Preparation</h4>
<div class="cell" data-execution_count="54">
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>nowotschin_metadata <span class="op">=</span> pd.read_table(<span class="st">"https://ftp.ncbi.nlm.nih.gov/geo/series/GSE123nnn/GSE123046/matrix/GSE123046_series_matrix.txt.gz"</span>,</span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>                                    skiprows<span class="op">=</span><span class="dv">41</span>, index_col<span class="op">=</span><span class="dv">0</span>).T</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="55">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>nowotschin_metadata[<span class="st">'ct'</span>] <span class="op">=</span> nowotschin_metadata.index</span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>nowotschin_metadata[<span class="st">'SRX'</span>] <span class="op">=</span> nowotschin_metadata[[<span class="st">'!Sample_relation'</span>]].agg(<span class="st">' '</span>.join, axis<span class="op">=</span><span class="dv">1</span>).<span class="bu">str</span>.extract(<span class="vs">r'(SRX[0-9]</span><span class="sc">{6}</span><span class="vs">)'</span>)</span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>nowotschin_metadata <span class="op">=</span> nowotschin_metadata.reset_index()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="56">
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a>nowotschin_metadata <span class="op">=</span> nowotschin_metadata[nowotschin_metadata.ct.<span class="bu">str</span>.startswith((<span class="st">"E3"</span>, <span class="st">"E4"</span>))].copy()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="57">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>nowotschin_metadata[<span class="st">'!Sample_geo_accession'</span>].to_csv(<span class="st">"../pipeline/fetchngs/GSE123046.txt"</span>, index<span class="op">=</span><span class="va">None</span>, header<span class="op">=</span><span class="va">None</span>)</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>nowotschin_metadata.set_index(<span class="st">'SRX'</span>).to_csv(<span class="st">"../pipeline/fetchngs/GSE123046_metadata.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-dataset-6" class="level4" data-number="1.7.2">
<h4 data-number="1.7.2" data-anchor-id="load-dataset-6"><span class="header-section-number">1.7.2</span> Load dataset</h4>
<div class="cell" data-execution_count="58">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> nowotschin_get_metadata():</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>    metadata <span class="op">=</span> pd.read_csv(<span class="st">"https://s3.amazonaws.com/dp-lab-data-public/mouse_endoderm/sc_endoderm_all_cells_metadata.csv"</span>, index_col<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>    correction <span class="op">=</span> pd.read_csv(<span class="st">"../data/external/aligned/mouse/Nowotschin_et_al/e35_cell_types.csv"</span>, index_col<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>    metadata.loc[correction.index, <span class="st">'CellType'</span>] <span class="op">=</span> correction.CellType</span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> metadata</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="59">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>nowotschin <span class="op">=</span> sc.read(<span class="st">"../data/external/aligned/mouse/Nowotschin_et_al/results/star/mtx_conversions/combined_matrix.h5ad"</span>)</span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>legend <span class="op">=</span> {</span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">'E4.5_Rep1'</span>: <span class="st">'Lib1-1_E4.5'</span>,</span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">'E4.5_Rep2'</span>: <span class="st">'Lib1-2_E4.5'</span>,</span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">'E3.5_Rep1'</span>: <span class="st">'Lib1-3_E3.5'</span>,</span>
<span id="cb80-7"><a href="#cb80-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">'E3.5_Rep2'</span>: <span class="st">'Lib1-4_E3.5'</span>,</span>
<span id="cb80-8"><a href="#cb80-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb80-9"><a href="#cb80-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-10"><a href="#cb80-10" aria-hidden="true" tabindex="-1"></a>nowotschin.obs[<span class="st">'Library'</span>] <span class="op">=</span> <span class="st">'UNKNOWN'</span></span>
<span id="cb80-11"><a href="#cb80-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> old_name, new_name <span class="kw">in</span> legend.items():</span>
<span id="cb80-12"><a href="#cb80-12" aria-hidden="true" tabindex="-1"></a>    nowotschin.obs.loc[nowotschin.obs[<span class="st">'sample_title'</span>] <span class="op">==</span> old_name, <span class="st">'Library'</span>] <span class="op">=</span> new_name</span>
<span id="cb80-13"><a href="#cb80-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-14"><a href="#cb80-14" aria-hidden="true" tabindex="-1"></a>nowotschin.obs[<span class="st">'cell_id'</span>] <span class="op">=</span> nowotschin.obs[<span class="st">'Library'</span>] <span class="op">+</span> <span class="st">"_"</span> <span class="op">+</span> nowotschin.obs_names.<span class="bu">str</span>.split(<span class="st">'_'</span>, expand<span class="op">=</span><span class="va">True</span>).droplevel(<span class="dv">1</span>)</span>
<span id="cb80-15"><a href="#cb80-15" aria-hidden="true" tabindex="-1"></a>nowotschin.obs[<span class="st">'timepoint'</span>] <span class="op">=</span> nowotschin.obs.sample_title.<span class="bu">str</span>.split(<span class="st">'_'</span>, expand<span class="op">=</span><span class="va">True</span>).values[:, <span class="dv">0</span>]</span>
<span id="cb80-16"><a href="#cb80-16" aria-hidden="true" tabindex="-1"></a>nowotschin.obs <span class="op">=</span> nowotschin.obs.set_index(<span class="st">'cell_id'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="60">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>nowotschin_metadata <span class="op">=</span> nowotschin_get_metadata()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="61">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>nowotschin <span class="op">=</span> nowotschin[nowotschin.obs_names.intersection(nowotschin_metadata.index)].copy()</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>nowotschin_metadata <span class="op">=</span> nowotschin_metadata.loc[nowotschin.obs_names].copy()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="62">
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>nowotschin.obs[<span class="st">'batch'</span>] <span class="op">=</span> <span class="st">"NOWO_1"</span></span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>nowotschin.obs.loc[nowotschin.obs[<span class="st">'timepoint'</span>].<span class="bu">str</span>.startswith(<span class="st">'E4.5'</span>), <span class="st">'batch'</span>] <span class="op">=</span> <span class="st">"NOWO_2"</span></span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a><span class="co"># nowotschin.obs['batch'] = nowotschin.obs['Library'].replace({</span></span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a><span class="co">#     'Lib1-1_E4.5': 'NOWO_1', </span></span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a><span class="co">#     'Lib1-2_E4.5': 'NOWO_2',</span></span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a><span class="co">#     'Lib1-3_E3.5': 'NOWO_3',</span></span>
<span id="cb83-7"><a href="#cb83-7" aria-hidden="true" tabindex="-1"></a><span class="co">#     'Lib1-4_E3.5': 'NOWO_4', </span></span>
<span id="cb83-8"><a href="#cb83-8" aria-hidden="true" tabindex="-1"></a><span class="co"># })</span></span>
<span id="cb83-9"><a href="#cb83-9" aria-hidden="true" tabindex="-1"></a>nowotschin.obs[<span class="st">'experiment'</span>] <span class="op">=</span> <span class="st">"Nowotschin et al., 2019"</span></span>
<span id="cb83-10"><a href="#cb83-10" aria-hidden="true" tabindex="-1"></a>nowotschin.obs[<span class="st">'technology'</span>] <span class="op">=</span> <span class="st">"10X 3' v2"</span></span>
<span id="cb83-11"><a href="#cb83-11" aria-hidden="true" tabindex="-1"></a>nowotschin.obs[<span class="st">'ct'</span>] <span class="op">=</span> nowotschin_metadata[[<span class="st">'Timepoint'</span>, <span class="st">'CellType'</span>]].agg(<span class="st">'-'</span>.join, axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb83-12"><a href="#cb83-12" aria-hidden="true" tabindex="-1"></a>nowotschin.obs[<span class="st">'ct_orig'</span>] <span class="op">=</span> nowotschin_metadata[<span class="st">'CellType'</span>]</span>
<span id="cb83-13"><a href="#cb83-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-14"><a href="#cb83-14" aria-hidden="true" tabindex="-1"></a>nowotschin.obs <span class="op">=</span> nowotschin.obs[[<span class="st">'batch'</span>, <span class="st">'experiment'</span>, <span class="st">'technology'</span>, <span class="st">'ct'</span>, <span class="st">'ct_orig'</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="fan-et-al.-2015-gse53386" class="level3" data-number="1.8">
<h3 data-number="1.8" data-anchor-id="fan-et-al.-2015-gse53386"><span class="header-section-number">1.8</span> Fan et al., 2015 <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE53386">GSE53386</a></h3>
<div class="sourceCode" id="cb84"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> <span class="dt">\</span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>    Fan_et_al <span class="dt">\</span></span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>    nextflow run nf-core/fetchngs <span class="dt">\</span></span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> 1.10.0 <span class="dt">\</span></span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">--force_sratools_download</span> <span class="dt">\</span></span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/fetchngs/GSE53386.txt</span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> Fan_et_al nextflow run brickmanlab/scrnaseq <span class="dt">\</span></span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> feature/smartseq <span class="dt">\</span></span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">-c</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/smartseq.config <span class="dt">\</span></span>
<span id="cb84-11"><a href="#cb84-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /scratch/Brickman/pipelines/Fan_et_al/results/samplesheet/samplesheet.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="preparation-7" class="level4" data-number="1.8.1">
<h4 data-number="1.8.1" data-anchor-id="preparation-7"><span class="header-section-number">1.8.1</span> Preparation</h4>
<div class="cell" data-execution_count="63">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>fan_metadata <span class="op">=</span> pd.read_table(<span class="st">"https://ftp.ncbi.nlm.nih.gov/geo/series/GSE53nnn/GSE53386/matrix/GSE53386-GPL13112_series_matrix.txt.gz"</span>, </span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a>                      skiprows<span class="op">=</span><span class="dv">33</span>, index_col<span class="op">=</span><span class="dv">0</span>).T</span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>fan_metadata[<span class="st">'SRX'</span>] <span class="op">=</span> fan_metadata[[<span class="st">'!Sample_relation'</span>]].agg(<span class="st">' '</span>.join, axis<span class="op">=</span><span class="dv">1</span>).<span class="bu">str</span>.extract(<span class="vs">r'(SRX[0-9]</span><span class="sc">{6}</span><span class="vs">)'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="64">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>fan_metadata[<span class="st">'ct'</span>] <span class="op">=</span> fan_metadata[<span class="st">'!Sample_characteristics_ch1'</span>].iloc[:, <span class="dv">0</span>].values</span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>fan_metadata <span class="op">=</span> fan_metadata.reset_index()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="65">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a>legend <span class="op">=</span> {</span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">'cell type: zygote'</span>: <span class="st">'Zygote'</span>,</span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">'cell type: 2-cell embryo'</span>: <span class="st">'2C'</span>,</span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">'cell type: 4-cell embryo'</span>: <span class="st">'4C'</span>,</span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">'cell type: 8-cell embryo'</span>: <span class="st">'8C'</span>, </span>
<span id="cb87-6"><a href="#cb87-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">'cell type: morula'</span>: <span class="st">'16C'</span>,</span>
<span id="cb87-7"><a href="#cb87-7" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="66">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> old_name, new_name <span class="kw">in</span> legend.items():</span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>    fan_metadata.loc[fan_metadata.ct <span class="op">==</span> old_name, <span class="st">'ct'</span>] <span class="op">=</span> new_name</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>fan_metadata <span class="op">=</span> fan_metadata[fan_metadata.ct.isin(legend.values())].copy()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="67">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>fan_metadata[<span class="st">'!Sample_geo_accession'</span>].to_csv(<span class="st">"../pipeline/fetchngs/GSE53386.txt"</span>, index<span class="op">=</span><span class="va">None</span>, header<span class="op">=</span><span class="va">None</span>)</span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>fan_metadata.set_index(<span class="st">'SRX'</span>).to_csv(<span class="st">"../pipeline/fetchngs/GSE53386_metadata.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-dataset-7" class="level4" data-number="1.8.2">
<h4 data-number="1.8.2" data-anchor-id="load-dataset-7"><span class="header-section-number">1.8.2</span> Load dataset</h4>
<div class="cell" data-execution_count="68">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>fan <span class="op">=</span> load_experiment(</span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../data/external/aligned/mouse/Fan_et_al/results/star/mtx_conversions/combined_matrix.h5ad"</span>,</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>    GEO<span class="op">=</span><span class="st">"GSE53386"</span></span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="69">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>fan.obs[<span class="st">'batch'</span>] <span class="op">=</span> <span class="st">"FAN_1"</span></span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>fan.obs[<span class="st">'experiment'</span>] <span class="op">=</span> <span class="st">"Fan et al., 2015"</span></span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>fan.obs[<span class="st">'technology'</span>] <span class="op">=</span> <span class="st">"SUPeR-seq"</span></span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-5"><a href="#cb91-5" aria-hidden="true" tabindex="-1"></a>fan.obs <span class="op">=</span> fan.obs[[<span class="st">'batch'</span>, <span class="st">'experiment'</span>, <span class="st">'technology'</span>, <span class="st">'ct'</span>, <span class="st">'ct_orig'</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="70">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a>fan <span class="op">=</span> normalize_smartseq(fan, genes_length)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>SMART-SEQ: Normalization
SMART-SEQ: Common genes 55364</code></pre>
</div>
</div>
<div class="cell" data-execution_count="71">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a>fan</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="71">
<pre><code>AnnData object with n_obs × n_vars = 27 × 55364
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig'
    var: 'gene_symbol'</code></pre>
</div>
</div>
</section>
</section>
<section id="borensztein-et-al.-2017-gse80810" class="level3" data-number="1.9">
<h3 data-number="1.9" data-anchor-id="borensztein-et-al.-2017-gse80810"><span class="header-section-number">1.9</span> Borensztein et al., 2017 <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE80810">GSE80810</a></h3>
<div class="sourceCode" id="cb96"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> <span class="dt">\</span></span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>    Borensztein_et_al <span class="dt">\</span></span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a>    nextflow run nf-core/fetchngs <span class="dt">\</span></span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> 1.10.0 <span class="dt">\</span></span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/fetchngs/GSE80810.txt</span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> Borensztein_et_al nextflow run brickmanlab/scrnaseq <span class="dt">\</span></span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> feature/smartseq <span class="dt">\</span></span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">-c</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/smartseq.config <span class="dt">\</span></span>
<span id="cb96-10"><a href="#cb96-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /scratch/Brickman/pipelines/Borensztein_et_al/results/samplesheet/samplesheet.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="preparation-8" class="level4" data-number="1.9.1">
<h4 data-number="1.9.1" data-anchor-id="preparation-8"><span class="header-section-number">1.9.1</span> Preparation</h4>
<div class="cell" data-execution_count="72">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>borenszrtein_metadata <span class="op">=</span> pd.read_table(<span class="st">"https://ftp.ncbi.nlm.nih.gov/geo/series/GSE80nnn/GSE80810/matrix/GSE80810_series_matrix.txt.gz"</span>,</span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>                                      skiprows<span class="op">=</span><span class="dv">31</span>, index_col<span class="op">=</span><span class="dv">0</span>).T</span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>borenszrtein_metadata[<span class="st">'SRX'</span>] <span class="op">=</span> borenszrtein_metadata[[<span class="st">'!Sample_relation'</span>]].iloc[:, :<span class="dv">2</span>].agg(<span class="st">' '</span>.join, axis<span class="op">=</span><span class="dv">1</span>).<span class="bu">str</span>.extract(<span class="vs">r'(SRX[0-9]</span><span class="sc">{6}</span><span class="vs">)'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="73">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>borenszrtein_metadata[<span class="st">'ct'</span>] <span class="op">=</span> borenszrtein_metadata[<span class="st">'!Sample_characteristics_ch1'</span>].iloc[:, <span class="dv">1</span>].values</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>borenszrtein_metadata <span class="op">=</span> borenszrtein_metadata.reset_index()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="74">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>legend <span class="op">=</span> {</span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 'developmental stage: Oocyte': 'Oocyte',</span></span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">'developmental stage: 2-cell'</span>: <span class="st">'2C'</span>,</span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">'developmental stage: 4-cell'</span>: <span class="st">'4C'</span>,</span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">'developmental stage: 8-cell'</span>: <span class="st">'8C'</span>,</span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">'developmental stage: 16-cell'</span>: <span class="st">'16C'</span>,</span>
<span id="cb99-7"><a href="#cb99-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 'developmental stage: 32-cell': 'UNKNOWN',</span></span>
<span id="cb99-8"><a href="#cb99-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 'developmental stage: 64-cell': 'UNKNOWN'</span></span>
<span id="cb99-9"><a href="#cb99-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb99-10"><a href="#cb99-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-11"><a href="#cb99-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> old_name, new_name <span class="kw">in</span> legend.items():</span>
<span id="cb99-12"><a href="#cb99-12" aria-hidden="true" tabindex="-1"></a>    borenszrtein_metadata.loc[borenszrtein_metadata.ct <span class="op">==</span> old_name, <span class="st">'ct'</span>] <span class="op">=</span> new_name</span>
<span id="cb99-13"><a href="#cb99-13" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb99-14"><a href="#cb99-14" aria-hidden="true" tabindex="-1"></a>borenszrtein_metadata <span class="op">=</span> borenszrtein_metadata[borenszrtein_metadata.ct.isin(legend.values())].copy()</span>
<span id="cb99-15"><a href="#cb99-15" aria-hidden="true" tabindex="-1"></a>borenszrtein_metadata <span class="op">=</span> borenszrtein_metadata[borenszrtein_metadata[<span class="st">'!Sample_characteristics_ch1'</span>].iloc[:, <span class="dv">2</span>].<span class="bu">str</span>.contains(<span class="st">'wt/wt'</span>)].copy()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="75">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>borenszrtein_metadata[<span class="st">'!Sample_geo_accession'</span>].to_csv(<span class="st">"../pipeline/fetchngs/GSE80810.txt"</span>, index<span class="op">=</span><span class="va">None</span>, header<span class="op">=</span><span class="va">None</span>)</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>borenszrtein_metadata.set_index(<span class="st">'SRX'</span>).to_csv(<span class="st">"../pipeline/fetchngs/GSE80810_metadata.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-dataset-8" class="level4" data-number="1.9.2">
<h4 data-number="1.9.2" data-anchor-id="load-dataset-8"><span class="header-section-number">1.9.2</span> Load dataset</h4>
<div class="cell" data-execution_count="76">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a>borensztein <span class="op">=</span> load_experiment(</span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../data/external/aligned/mouse/Borensztein_et_al/results/star/mtx_conversions/combined_matrix.h5ad"</span>,</span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a>    GEO<span class="op">=</span><span class="st">"GSE80810"</span>, left_on<span class="op">=</span><span class="st">'sample_alias'</span>, right_on<span class="op">=</span><span class="st">'ID_REF'</span></span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="77">
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a>borensztein.obs[<span class="st">'batch'</span>] <span class="op">=</span> <span class="st">"BORENSZTEIN_1"</span></span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>borensztein.obs[<span class="st">'experiment'</span>] <span class="op">=</span> <span class="st">"Borensztein et al., 2017"</span></span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>borensztein.obs[<span class="st">'technology'</span>] <span class="op">=</span> <span class="st">"qRT-PCR"</span></span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a>borensztein.obs <span class="op">=</span> borensztein.obs[[<span class="st">'batch'</span>, <span class="st">'experiment'</span>, <span class="st">'technology'</span>, <span class="st">'ct'</span>, <span class="st">'ct_orig'</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="78">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a>borensztein <span class="op">=</span> normalize_smartseq(borensztein, genes_length)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>SMART-SEQ: Normalization
SMART-SEQ: Common genes 55364</code></pre>
</div>
</div>
<div class="cell" data-execution_count="79">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a>borensztein</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="79">
<pre><code>AnnData object with n_obs × n_vars = 60 × 55364
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig'
    var: 'gene_symbol'</code></pre>
</div>
</div>
</section>
</section>
<section id="stirparo-et-al.-2020-gse159030" class="level3" data-number="1.10">
<h3 data-number="1.10" data-anchor-id="stirparo-et-al.-2020-gse159030"><span class="header-section-number">1.10</span> Stirparo et al., 2020 <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE159030">GSE159030</a></h3>
<div class="sourceCode" id="cb107"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> <span class="dt">\</span></span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a>    Stirparo_et_al <span class="dt">\</span></span>
<span id="cb107-3"><a href="#cb107-3" aria-hidden="true" tabindex="-1"></a>    nextflow run nf-core/fetchngs <span class="dt">\</span></span>
<span id="cb107-4"><a href="#cb107-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> 1.10.0 <span class="dt">\</span></span>
<span id="cb107-5"><a href="#cb107-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">--force_sratools_download</span> <span class="dt">\</span></span>
<span id="cb107-6"><a href="#cb107-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/fetchngs/GSE159030.txt</span>
<span id="cb107-7"><a href="#cb107-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-8"><a href="#cb107-8" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> Stirparo_et_al nextflow run brickmanlab/scrnaseq <span class="dt">\</span></span>
<span id="cb107-9"><a href="#cb107-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> feature/smartseq <span class="dt">\</span></span>
<span id="cb107-10"><a href="#cb107-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">-c</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/smartseq.config <span class="dt">\</span></span>
<span id="cb107-11"><a href="#cb107-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /scratch/Brickman/pipelines/Stirparo_et_al/results/samplesheet/samplesheet.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="preparation-9" class="level4" data-number="1.10.1">
<h4 data-number="1.10.1" data-anchor-id="preparation-9"><span class="header-section-number">1.10.1</span> Preparation</h4>
<div class="cell" data-execution_count="80">
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a>stirparo_metadata <span class="op">=</span> pd.read_table(<span class="st">"https://ftp.ncbi.nlm.nih.gov/geo/series/GSE159nnn/GSE159030/matrix/GSE159030_series_matrix.txt.gz"</span>,</span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a>                                  skiprows<span class="op">=</span><span class="dv">27</span>, index_col<span class="op">=</span><span class="dv">0</span>).T</span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a>stirparo_metadata[<span class="st">'SRX'</span>] <span class="op">=</span> stirparo_metadata[[<span class="st">'!Sample_relation'</span>]].agg(<span class="st">' '</span>.join, axis<span class="op">=</span><span class="dv">1</span>).<span class="bu">str</span>.extract(<span class="vs">r'(SRX[0-9]</span><span class="sc">{6}</span><span class="vs">)'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="81">
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a>stirparo_metadata[<span class="st">'ct'</span>] <span class="op">=</span> stirparo_metadata[<span class="st">'!Sample_source_name_ch1'</span>]</span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a>stirparo_metadata <span class="op">=</span> stirparo_metadata.reset_index()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="82">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a>legend <span class="op">=</span> {</span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Primitive Endoderm'</span>: <span class="st">'PrE'</span>,</span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">'Epiblast'</span>: <span class="st">'EPI'</span>,</span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb110-5"><a href="#cb110-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-6"><a href="#cb110-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> old_name, new_name <span class="kw">in</span> legend.items():</span>
<span id="cb110-7"><a href="#cb110-7" aria-hidden="true" tabindex="-1"></a>    stirparo_metadata.loc[stirparo_metadata.ct <span class="op">==</span> old_name, <span class="st">'ct'</span>] <span class="op">=</span> new_name</span>
<span id="cb110-8"><a href="#cb110-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-9"><a href="#cb110-9" aria-hidden="true" tabindex="-1"></a>stirparo_metadata[<span class="st">'ct'</span>] <span class="op">=</span> stirparo_metadata[<span class="st">'!Sample_characteristics_ch1'</span>].iloc[:, <span class="dv">1</span>].<span class="bu">str</span>.replace(<span class="st">'developmental stage: '</span>, <span class="st">''</span>).astype(<span class="bu">str</span>)<span class="op">\</span></span>
<span id="cb110-10"><a href="#cb110-10" aria-hidden="true" tabindex="-1"></a>                        <span class="op">+</span> <span class="st">'-'</span> <span class="op">+</span> stirparo_metadata.ct</span>
<span id="cb110-11"><a href="#cb110-11" aria-hidden="true" tabindex="-1"></a>stirparo_metadata <span class="op">=</span> stirparo_metadata[stirparo_metadata[<span class="st">'!Sample_characteristics_ch1'</span>].iloc[:, <span class="dv">0</span>] <span class="op">==</span> <span class="st">'genotype: Wild-Type'</span>].copy()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="83">
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>stirparo_metadata[<span class="st">'!Sample_geo_accession'</span>].to_csv(<span class="st">"../pipeline/fetchngs/GSE159030.txt"</span>, index<span class="op">=</span><span class="va">None</span>, header<span class="op">=</span><span class="va">None</span>)</span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a>stirparo_metadata.set_index(<span class="st">'SRX'</span>).to_csv(<span class="st">"../pipeline/fetchngs/GSE159030_metadata.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-dataset-9" class="level4" data-number="1.10.2">
<h4 data-number="1.10.2" data-anchor-id="load-dataset-9"><span class="header-section-number">1.10.2</span> Load dataset</h4>
<div class="cell" data-execution_count="84">
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a>stirparo <span class="op">=</span> load_experiment(</span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../data/external/aligned/mouse/Stirparo_et_al/results/star/mtx_conversions/combined_matrix.h5ad"</span>,</span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a>    GEO<span class="op">=</span><span class="st">"GSE159030"</span>, left_on<span class="op">=</span><span class="st">'sample_alias'</span>, right_on<span class="op">=</span><span class="st">'ID_REF'</span></span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="85">
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a>stirparo.obs[<span class="st">'batch'</span>] <span class="op">=</span> <span class="st">"STIRPARO_1"</span></span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>stirparo.obs[<span class="st">'experiment'</span>] <span class="op">=</span> <span class="st">"Stirparo et al., 2020"</span></span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>stirparo.obs[<span class="st">'technology'</span>] <span class="op">=</span> <span class="st">"SMART-seq2"</span></span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a>stirparo.obs <span class="op">=</span> stirparo.obs[[<span class="st">'batch'</span>, <span class="st">'experiment'</span>, <span class="st">'technology'</span>, <span class="st">'ct'</span>, <span class="st">'ct_orig'</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="86">
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a>stirparo <span class="op">=</span> normalize_smartseq(stirparo, genes_length)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>SMART-SEQ: Normalization
SMART-SEQ: Common genes 55364</code></pre>
</div>
</div>
<div class="cell" data-execution_count="87">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a>stirparo</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="87">
<pre><code>AnnData object with n_obs × n_vars = 64 × 55364
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig'
    var: 'gene_symbol'</code></pre>
</div>
</div>
</section>
</section>
<section id="xue-et-al.-2013-gse44183" class="level3" data-number="1.11">
<h3 data-number="1.11" data-anchor-id="xue-et-al.-2013-gse44183"><span class="header-section-number">1.11</span> Xue et al., 2013 <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE44183">GSE44183</a></h3>
<div class="sourceCode" id="cb118"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> <span class="dt">\</span></span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a>    Xue_et_al <span class="dt">\</span></span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a>    nextflow run nf-core/fetchngs <span class="dt">\</span></span>
<span id="cb118-4"><a href="#cb118-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> 1.10.0 <span class="dt">\</span></span>
<span id="cb118-5"><a href="#cb118-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/fetchngs/GSE44183.txt</span>
<span id="cb118-6"><a href="#cb118-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-7"><a href="#cb118-7" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> Xue_et_al nextflow run brickmanlab/scrnaseq <span class="dt">\</span></span>
<span id="cb118-8"><a href="#cb118-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> feature/smartseq <span class="dt">\</span></span>
<span id="cb118-9"><a href="#cb118-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">-c</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/smartseq.config <span class="dt">\</span></span>
<span id="cb118-10"><a href="#cb118-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /scratch/Brickman/pipelines/Xue_et_al/results/samplesheet/samplesheet.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="preparation-10" class="level4" data-number="1.11.1">
<h4 data-number="1.11.1" data-anchor-id="preparation-10"><span class="header-section-number">1.11.1</span> Preparation</h4>
<div class="cell" data-execution_count="88">
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a>xue_metadata <span class="op">=</span> pd.read_table(<span class="st">"https://ftp.ncbi.nlm.nih.gov/geo/series/GSE44nnn/GSE44183/matrix/GSE44183-GPL13112_series_matrix.txt.gz"</span>,</span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a>                               skiprows<span class="op">=</span><span class="dv">35</span>, index_col<span class="op">=</span><span class="dv">0</span>).T</span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a>xue_metadata[<span class="st">'SRX'</span>] <span class="op">=</span> xue_metadata[[<span class="st">'!Sample_relation'</span>]].agg(<span class="st">' '</span>.join, axis<span class="op">=</span><span class="dv">1</span>).<span class="bu">str</span>.extract(<span class="vs">r'(SRX[0-9]</span><span class="sc">{6}</span><span class="vs">)'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="89">
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a>xue_metadata[<span class="st">'ct'</span>] <span class="op">=</span> xue_metadata[<span class="st">'!Sample_source_name_ch1'</span>]</span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a>xue_metadata <span class="op">=</span> xue_metadata.reset_index()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="90">
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a>legend <span class="op">=</span> {</span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 'oocyte', : '',</span></span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 'pronucleus': '',</span></span>
<span id="cb121-4"><a href="#cb121-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">'2-cell blastomere'</span>: <span class="st">'2C'</span>, </span>
<span id="cb121-5"><a href="#cb121-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">'4-cell blastomere'</span>: <span class="st">'4C'</span>,</span>
<span id="cb121-6"><a href="#cb121-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">'8-cell blastomere'</span>: <span class="st">'8C'</span>, </span>
<span id="cb121-7"><a href="#cb121-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">'morula'</span>: <span class="st">'16C'</span></span>
<span id="cb121-8"><a href="#cb121-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb121-9"><a href="#cb121-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-10"><a href="#cb121-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> old_name, new_name <span class="kw">in</span> legend.items():</span>
<span id="cb121-11"><a href="#cb121-11" aria-hidden="true" tabindex="-1"></a>    xue_metadata.loc[xue_metadata.ct <span class="op">==</span> old_name, <span class="st">'ct'</span>] <span class="op">=</span> new_name</span>
<span id="cb121-12"><a href="#cb121-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-13"><a href="#cb121-13" aria-hidden="true" tabindex="-1"></a>xue_metadata <span class="op">=</span> xue_metadata[xue_metadata.ct.isin(legend.values())].copy()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="91">
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a>xue_metadata[<span class="st">'!Sample_geo_accession'</span>].to_csv(<span class="st">"../pipeline/fetchngs/GSE44183.txt"</span>, index<span class="op">=</span><span class="va">None</span>, header<span class="op">=</span><span class="va">None</span>)</span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a>xue_metadata.set_index(<span class="st">'SRX'</span>).to_csv(<span class="st">"../pipeline/fetchngs/GSE44183_metadata.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-dataset-10" class="level4" data-number="1.11.2">
<h4 data-number="1.11.2" data-anchor-id="load-dataset-10"><span class="header-section-number">1.11.2</span> Load dataset</h4>
<div class="cell" data-execution_count="92">
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a>xue <span class="op">=</span> load_experiment(</span>
<span id="cb123-2"><a href="#cb123-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../data/external/aligned/mouse/Xue_et_al/results/star/mtx_conversions/combined_matrix.h5ad"</span>,</span>
<span id="cb123-3"><a href="#cb123-3" aria-hidden="true" tabindex="-1"></a>    GEO<span class="op">=</span><span class="st">"GSE44183"</span></span>
<span id="cb123-4"><a href="#cb123-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="93">
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a>xue.obs[<span class="st">'batch'</span>] <span class="op">=</span> <span class="st">"XUE_1"</span></span>
<span id="cb124-2"><a href="#cb124-2" aria-hidden="true" tabindex="-1"></a>xue.obs[<span class="st">'experiment'</span>] <span class="op">=</span> <span class="st">"Xue et al., 2013"</span></span>
<span id="cb124-3"><a href="#cb124-3" aria-hidden="true" tabindex="-1"></a>xue.obs[<span class="st">'technology'</span>] <span class="op">=</span> <span class="st">"qRT-PCR"</span></span>
<span id="cb124-4"><a href="#cb124-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-5"><a href="#cb124-5" aria-hidden="true" tabindex="-1"></a>xue.obs <span class="op">=</span> xue.obs[[<span class="st">'batch'</span>, <span class="st">'experiment'</span>, <span class="st">'technology'</span>, <span class="st">'ct'</span>, <span class="st">'ct_orig'</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="94">
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a>xue <span class="op">=</span> normalize_smartseq(xue, genes_length)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>SMART-SEQ: Normalization
SMART-SEQ: Common genes 55364</code></pre>
</div>
</div>
<div class="cell" data-execution_count="95">
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a>xue</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="95">
<pre><code>AnnData object with n_obs × n_vars = 12 × 55364
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig'
    var: 'gene_symbol'</code></pre>
</div>
</div>
</section>
</section>
<section id="yanagida-et-al.-2022-gse148462" class="level3" data-number="1.12">
<h3 data-number="1.12" data-anchor-id="yanagida-et-al.-2022-gse148462"><span class="header-section-number">1.12</span> Yanagida et al., 2022 <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE148462">GSE148462</a></h3>
<div class="sourceCode" id="cb129"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> <span class="dt">\</span></span>
<span id="cb129-2"><a href="#cb129-2" aria-hidden="true" tabindex="-1"></a>    Yanagida_et_al <span class="dt">\</span></span>
<span id="cb129-3"><a href="#cb129-3" aria-hidden="true" tabindex="-1"></a>    nextflow run nf-core/fetchngs <span class="dt">\</span></span>
<span id="cb129-4"><a href="#cb129-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> 1.10.0 <span class="dt">\</span></span>
<span id="cb129-5"><a href="#cb129-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">--force_sratools_download</span> <span class="dt">\</span></span>
<span id="cb129-6"><a href="#cb129-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/fetchngs/GSE148462.txt</span>
<span id="cb129-7"><a href="#cb129-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-8"><a href="#cb129-8" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> Yanagida_et_al nextflow run brickmanlab/scrnaseq <span class="dt">\</span></span>
<span id="cb129-9"><a href="#cb129-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> feature/smartseq <span class="dt">\</span></span>
<span id="cb129-10"><a href="#cb129-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">-c</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/smartseq.config <span class="dt">\</span></span>
<span id="cb129-11"><a href="#cb129-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /scratch/Brickman/pipelines/Yanagida_et_al/results/samplesheet/samplesheet.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="preparation-11" class="level4" data-number="1.12.1">
<h4 data-number="1.12.1" data-anchor-id="preparation-11"><span class="header-section-number">1.12.1</span> Preparation</h4>
<div class="cell" data-execution_count="96">
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a>yanagida_metadata <span class="op">=</span> pd.read_table(<span class="st">"https://ftp.ncbi.nlm.nih.gov/geo/series/GSE148nnn/GSE148462/matrix/GSE148462_series_matrix.txt.gz"</span>,</span>
<span id="cb130-2"><a href="#cb130-2" aria-hidden="true" tabindex="-1"></a>                                  skiprows<span class="op">=</span><span class="dv">29</span>, index_col<span class="op">=</span><span class="dv">0</span>).T</span>
<span id="cb130-3"><a href="#cb130-3" aria-hidden="true" tabindex="-1"></a>yanagida_metadata[<span class="st">'SRX'</span>] <span class="op">=</span> yanagida_metadata[[<span class="st">'!Sample_relation'</span>]].agg(<span class="st">' '</span>.join, axis<span class="op">=</span><span class="dv">1</span>).<span class="bu">str</span>.extract(<span class="vs">r'(SRX[0-9]</span><span class="sc">{6}</span><span class="vs">)'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="97">
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a>yanagida_metadata[<span class="st">'ct'</span>] <span class="op">=</span> <span class="st">'E3.75-ICM'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="98">
<div class="sourceCode cell-code" id="cb132"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb132-1"><a href="#cb132-1" aria-hidden="true" tabindex="-1"></a>yanagida_metadata[<span class="st">'!Sample_geo_accession'</span>].to_csv(<span class="st">"../pipeline/fetchngs/GSE148462.txt"</span>, index<span class="op">=</span><span class="va">None</span>, header<span class="op">=</span><span class="va">None</span>)</span>
<span id="cb132-2"><a href="#cb132-2" aria-hidden="true" tabindex="-1"></a>yanagida_metadata.set_index(<span class="st">'SRX'</span>).to_csv(<span class="st">"../pipeline/fetchngs/GSE148462_metadata.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-dataset-11" class="level4" data-number="1.12.2">
<h4 data-number="1.12.2" data-anchor-id="load-dataset-11"><span class="header-section-number">1.12.2</span> Load dataset</h4>
<div class="cell" data-execution_count="99">
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a>yanagida <span class="op">=</span> load_experiment(</span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../data/external/aligned/mouse/Yanagida_et_al/results/star/mtx_conversions/combined_matrix.h5ad"</span>,</span>
<span id="cb133-3"><a href="#cb133-3" aria-hidden="true" tabindex="-1"></a>    GEO<span class="op">=</span><span class="st">"GSE148462"</span>, left_on<span class="op">=</span><span class="st">'sample_alias'</span>, right_on<span class="op">=</span><span class="st">'ID_REF'</span></span>
<span id="cb133-4"><a href="#cb133-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="100">
<div class="sourceCode cell-code" id="cb134"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb134-1"><a href="#cb134-1" aria-hidden="true" tabindex="-1"></a>yanagida.obs[<span class="st">'batch'</span>] <span class="op">=</span> <span class="st">"YANAGIDA_1"</span></span>
<span id="cb134-2"><a href="#cb134-2" aria-hidden="true" tabindex="-1"></a>yanagida.obs[<span class="st">'experiment'</span>] <span class="op">=</span> <span class="st">"Yanagida et al., 2022"</span></span>
<span id="cb134-3"><a href="#cb134-3" aria-hidden="true" tabindex="-1"></a>yanagida.obs[<span class="st">'technology'</span>] <span class="op">=</span> <span class="st">"SMART-seq2"</span></span>
<span id="cb134-4"><a href="#cb134-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-5"><a href="#cb134-5" aria-hidden="true" tabindex="-1"></a>yanagida.obs <span class="op">=</span> yanagida.obs[[<span class="st">'batch'</span>, <span class="st">'experiment'</span>, <span class="st">'technology'</span>, <span class="st">'ct'</span>, <span class="st">'ct_orig'</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="101">
<div class="sourceCode cell-code" id="cb135"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a>yanagida <span class="op">=</span> normalize_smartseq(yanagida, genes_length)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>SMART-SEQ: Normalization
SMART-SEQ: Common genes 55364</code></pre>
</div>
</div>
<div class="cell" data-execution_count="102">
<div class="sourceCode cell-code" id="cb137"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb137-1"><a href="#cb137-1" aria-hidden="true" tabindex="-1"></a>yanagida</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="102">
<pre><code>AnnData object with n_obs × n_vars = 48 × 55364
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig'
    var: 'gene_symbol'</code></pre>
</div>
</div>
</section>
</section>
</section>
<section id="mohammed-et-al.-2017-gse100597" class="level2" data-number="2">
<h2 data-number="2" data-anchor-id="mohammed-et-al.-2017-gse100597"><span class="header-section-number">2</span> Mohammed et al., 2017 <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE100597">GSE100597</a></h2>
<div class="sourceCode" id="cb139"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> <span class="dt">\</span></span>
<span id="cb139-2"><a href="#cb139-2" aria-hidden="true" tabindex="-1"></a>    Mohammed_et_al <span class="dt">\</span></span>
<span id="cb139-3"><a href="#cb139-3" aria-hidden="true" tabindex="-1"></a>    nextflow run nf-core/fetchngs <span class="dt">\</span></span>
<span id="cb139-4"><a href="#cb139-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> 1.10.0 <span class="dt">\</span></span>
<span id="cb139-5"><a href="#cb139-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/fetchngs/GSE100597.txt</span>
<span id="cb139-6"><a href="#cb139-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-7"><a href="#cb139-7" aria-hidden="true" tabindex="-1"></a><span class="ex">nf-core_tower.sh</span> Mohammed_et_al nextflow run brickmanlab/scrnaseq <span class="dt">\</span></span>
<span id="cb139-8"><a href="#cb139-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">-r</span> feature/smartseq <span class="dt">\</span></span>
<span id="cb139-9"><a href="#cb139-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">-c</span> /projects/dan1/data/Brickman/projects/proks-salehin-et-al-2023/pipeline/smartseq.config <span class="dt">\</span></span>
<span id="cb139-10"><a href="#cb139-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">--input</span> /scratch/Brickman/pipelines/Mohammed_et_al/results/samplesheet/samplesheet.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="preparation-12" class="level3" data-number="2.1">
<h3 data-number="2.1" data-anchor-id="preparation-12"><span class="header-section-number">2.1</span> Preparation</h3>
<div class="cell" data-execution_count="103">
<div class="sourceCode cell-code" id="cb140"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb140-1"><a href="#cb140-1" aria-hidden="true" tabindex="-1"></a>mohammed_metadata <span class="op">=</span> pd.read_table(<span class="st">"https://ftp.ncbi.nlm.nih.gov/geo/series/GSE100nnn/GSE100597/matrix/GSE100597_series_matrix.txt.gz"</span>,</span>
<span id="cb140-2"><a href="#cb140-2" aria-hidden="true" tabindex="-1"></a>                                  skiprows<span class="op">=</span><span class="dv">27</span>, index_col<span class="op">=</span><span class="dv">0</span>).T</span>
<span id="cb140-3"><a href="#cb140-3" aria-hidden="true" tabindex="-1"></a>mohammed_metadata[<span class="st">'SRX'</span>] <span class="op">=</span> mohammed_metadata[[<span class="st">'!Sample_relation'</span>]].agg(<span class="st">' '</span>.join, axis<span class="op">=</span><span class="dv">1</span>).<span class="bu">str</span>.extract(<span class="vs">r'(SRX[0-9]</span><span class="sc">{6}</span><span class="vs">)'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="104">
<div class="sourceCode cell-code" id="cb141"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb141-1"><a href="#cb141-1" aria-hidden="true" tabindex="-1"></a>mohammed_metadata <span class="op">=</span> mohammed_metadata[mohammed_metadata[<span class="st">'!Sample_source_name_ch1'</span>].isin([<span class="st">'E3.5'</span>, <span class="st">'E4.5'</span>])]</span>
<span id="cb141-2"><a href="#cb141-2" aria-hidden="true" tabindex="-1"></a>mohammed_metadata <span class="op">=</span> mohammed_metadata[mohammed_metadata.index.<span class="bu">str</span>.endswith(<span class="st">'single'</span>)].copy()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="105">
<div class="sourceCode cell-code" id="cb142"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a>mohammed_metadata[<span class="st">'!Sample_geo_accession'</span>].to_csv(<span class="st">"../pipeline/fetchngs/GSE100597.txt"</span>, index<span class="op">=</span><span class="va">None</span>, header<span class="op">=</span><span class="va">None</span>)</span>
<span id="cb142-2"><a href="#cb142-2" aria-hidden="true" tabindex="-1"></a>mohammed_metadata.set_index(<span class="st">'SRX'</span>).to_csv(<span class="st">"../pipeline/fetchngs/GSE100597_metadata.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-data" class="level3" data-number="2.2">
<h3 data-number="2.2" data-anchor-id="load-data"><span class="header-section-number">2.2</span> Load data</h3>
<div class="cell" data-execution_count="106">
<div class="sourceCode cell-code" id="cb143"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb143-1"><a href="#cb143-1" aria-hidden="true" tabindex="-1"></a>mohammed <span class="op">=</span> load_experiment(</span>
<span id="cb143-2"><a href="#cb143-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../data/external/aligned/mouse/Mohammed_et_al/results/star/mtx_conversions/combined_matrix.h5ad"</span>,</span>
<span id="cb143-3"><a href="#cb143-3" aria-hidden="true" tabindex="-1"></a>    GEO<span class="op">=</span><span class="st">"GSE100597"</span>, left_on<span class="op">=</span><span class="st">'sample_alias'</span>, right_on<span class="op">=</span><span class="st">'ID_REF'</span></span>
<span id="cb143-4"><a href="#cb143-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="107">
<div class="sourceCode cell-code" id="cb144"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb144-1"><a href="#cb144-1" aria-hidden="true" tabindex="-1"></a>mohammed_metadata <span class="op">=</span> pd.read_table(<span class="st">"../data/external/aligned/mouse/Mohammed_et_al/Cell_clusters.txt"</span>, index_col<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb144-2"><a href="#cb144-2" aria-hidden="true" tabindex="-1"></a>mohammed_metadata.Lineage <span class="op">=</span> mohammed_metadata.Lineage.<span class="bu">str</span>.replace(<span class="st">'epiblast'</span>, <span class="st">'EPI'</span>).<span class="bu">str</span>.replace(<span class="st">'PE'</span>, <span class="st">'PrE'</span>)</span>
<span id="cb144-3"><a href="#cb144-3" aria-hidden="true" tabindex="-1"></a>mohammed.obs[<span class="st">'ct'</span>] <span class="op">=</span> mohammed_metadata.loc[mohammed.obs.ct_orig, [<span class="st">'Stage'</span>, <span class="st">'Lineage'</span>]].agg(<span class="st">'-'</span>.join, axis<span class="op">=</span><span class="dv">1</span>).values</span>
<span id="cb144-4"><a href="#cb144-4" aria-hidden="true" tabindex="-1"></a>mohammed <span class="op">=</span> mohammed[mohammed.obs.ct <span class="op">!=</span> <span class="st">'E4.5-interm'</span>].copy()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="108">
<div class="sourceCode cell-code" id="cb145"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a>mohammed.obs[<span class="st">'batch'</span>] <span class="op">=</span> <span class="st">"MOHAMMED_1"</span></span>
<span id="cb145-2"><a href="#cb145-2" aria-hidden="true" tabindex="-1"></a>mohammed.obs[<span class="st">'experiment'</span>] <span class="op">=</span> <span class="st">"Mohammed et al., 2017"</span></span>
<span id="cb145-3"><a href="#cb145-3" aria-hidden="true" tabindex="-1"></a>mohammed.obs[<span class="st">'technology'</span>] <span class="op">=</span> <span class="st">"SMART-seq2"</span></span>
<span id="cb145-4"><a href="#cb145-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb145-5"><a href="#cb145-5" aria-hidden="true" tabindex="-1"></a>mohammed.obs <span class="op">=</span> mohammed.obs[[<span class="st">'batch'</span>, <span class="st">'experiment'</span>, <span class="st">'technology'</span>, <span class="st">'ct'</span>, <span class="st">'ct_orig'</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="109">
<div class="sourceCode cell-code" id="cb146"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb146-1"><a href="#cb146-1" aria-hidden="true" tabindex="-1"></a>mohammed <span class="op">=</span> normalize_smartseq(mohammed, genes_length)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>SMART-SEQ: Normalization
SMART-SEQ: Common genes 55364</code></pre>
</div>
</div>
<div class="cell" data-execution_count="110">
<div class="sourceCode cell-code" id="cb148"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a>mohammed</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="110">
<pre><code>AnnData object with n_obs × n_vars = 153 × 55364
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig'
    var: 'gene_symbol'</code></pre>
</div>
</div>
</section>
</section>
<section id="integration" class="level2" data-number="3">
<h2 data-number="3" data-anchor-id="integration"><span class="header-section-number">3</span> Integration</h2>
<div class="cell" data-execution_count="111">
<div class="sourceCode cell-code" id="cb150"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb150-1"><a href="#cb150-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> urllib.request, json</span>
<span id="cb150-2"><a href="#cb150-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-3"><a href="#cb150-3" aria-hidden="true" tabindex="-1"></a>cc_url <span class="op">=</span> <span class="st">"https://github.com/brickmanlab/project-template/raw/master/%7B%7B</span><span class="sc">%20c</span><span class="st">ookiecutter.project_name</span><span class="sc">%20%</span><span class="st">7D%7D/data/external/mouse_cell_cycle_genes.json"</span></span>
<span id="cb150-4"><a href="#cb150-4" aria-hidden="true" tabindex="-1"></a><span class="cf">with</span> urllib.request.urlopen(cc_url) <span class="im">as</span> url:</span>
<span id="cb150-5"><a href="#cb150-5" aria-hidden="true" tabindex="-1"></a>    cc_dict <span class="op">=</span> json.load(url)</span>
<span id="cb150-6"><a href="#cb150-6" aria-hidden="true" tabindex="-1"></a>    cc <span class="op">=</span> <span class="bu">sum</span>(<span class="bu">list</span>(cc_dict.values()), [])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="112">
<div class="sourceCode cell-code" id="cb151"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a>adata <span class="op">=</span> anndata.concat([deng, biase, posfai, goolam, boroviak, chen, nowotschin, fan, borensztein, stirparo, xue, yanagida, mohammed])</span>
<span id="cb151-2"><a href="#cb151-2" aria-hidden="true" tabindex="-1"></a>adata.obs[<span class="st">'stage'</span>] <span class="op">=</span> adata.obs.ct.<span class="bu">str</span>.replace(<span class="st">'E3.25-|E3.5-|E3.75-|E4.5-'</span>, <span class="st">''</span>, regex<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb151-3"><a href="#cb151-3" aria-hidden="true" tabindex="-1"></a>adata.obs[<span class="st">'timepoint'</span>] <span class="op">=</span> adata.obs.ct.<span class="bu">str</span>.split(<span class="st">'-'</span>, expand<span class="op">=</span><span class="va">True</span>).iloc[:, <span class="dv">0</span>]</span>
<span id="cb151-4"><a href="#cb151-4" aria-hidden="true" tabindex="-1"></a>adata.obs.ct <span class="op">=</span> pd.Categorical(adata.obs.ct, </span>
<span id="cb151-5"><a href="#cb151-5" aria-hidden="true" tabindex="-1"></a>                              categories<span class="op">=</span>[<span class="st">'Zygote'</span>, <span class="st">'2C'</span>, <span class="st">'4C'</span>, <span class="st">'8C'</span>, <span class="st">'16C'</span>, </span>
<span id="cb151-6"><a href="#cb151-6" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">'E3.25-ICM'</span>, <span class="st">'E3.25-TE'</span>, </span>
<span id="cb151-7"><a href="#cb151-7" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">'E3.5-ICM'</span>, <span class="st">'E3.5-TE'</span>, <span class="st">'E3.5-EPI'</span>, <span class="st">'E3.5-PrE'</span>, </span>
<span id="cb151-8"><a href="#cb151-8" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">'E3.75-ICM'</span>, </span>
<span id="cb151-9"><a href="#cb151-9" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">'E4.5-TE'</span>, <span class="st">'E4.5-EPI'</span>, <span class="st">'E4.5-PrE'</span>],</span>
<span id="cb151-10"><a href="#cb151-10" aria-hidden="true" tabindex="-1"></a>                              ordered<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb151-11"><a href="#cb151-11" aria-hidden="true" tabindex="-1"></a>adata.obs_names_make_unique()</span>
<span id="cb151-12"><a href="#cb151-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-13"><a href="#cb151-13" aria-hidden="true" tabindex="-1"></a>adata.var[<span class="st">'gene_ids'</span>] <span class="op">=</span> adata.var_names</span>
<span id="cb151-14"><a href="#cb151-14" aria-hidden="true" tabindex="-1"></a>adata.var[<span class="st">'gene_symbol'</span>] <span class="op">=</span> deng[:, adata.var_names].var[<span class="st">'gene_symbol'</span>].values</span>
<span id="cb151-15"><a href="#cb151-15" aria-hidden="true" tabindex="-1"></a>adata.var.index <span class="op">=</span> adata.var.gene_symbol.<span class="bu">str</span>.lower().values</span>
<span id="cb151-16"><a href="#cb151-16" aria-hidden="true" tabindex="-1"></a>adata.var_names_make_unique()</span>
<span id="cb151-17"><a href="#cb151-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-18"><a href="#cb151-18" aria-hidden="true" tabindex="-1"></a><span class="co"># remove ribosomal genes</span></span>
<span id="cb151-19"><a href="#cb151-19" aria-hidden="true" tabindex="-1"></a>adata <span class="op">=</span> adata[:, <span class="op">~</span>np.logical_or(adata.var_names.<span class="bu">str</span>.startswith((<span class="st">'rps'</span>,<span class="st">'rpl'</span>)), adata.var_names <span class="op">==</span> <span class="st">"ct010467.1"</span>)].copy()</span>
<span id="cb151-20"><a href="#cb151-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-21"><a href="#cb151-21" aria-hidden="true" tabindex="-1"></a><span class="co"># remove cc genes</span></span>
<span id="cb151-22"><a href="#cb151-22" aria-hidden="true" tabindex="-1"></a>adata <span class="op">=</span> adata[:, <span class="op">~</span>adata.var_names.isin(cc)].copy()</span>
<span id="cb151-23"><a href="#cb151-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-24"><a href="#cb151-24" aria-hidden="true" tabindex="-1"></a>adata</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="112">
<pre><code>AnnData object with n_obs × n_vars = 2022 × 54940
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig', 'stage', 'timepoint'
    var: 'gene_ids', 'gene_symbol'</code></pre>
</div>
</div>
<div class="cell" data-execution_count="113">
<div class="sourceCode cell-code" id="cb153"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a>get_summary(adata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Number of MITO genes: 0
Number of ERCC genes: 0
Number of RIBO genes: 0</code></pre>
</div>
<section id="number-of-cells-per-experiment" class="level3 cell-output cell-output-display" data-number="3.1">
<h3 data-number="3.1" data-anchor-id="number-of-cells-per-experiment"><span class="header-section-number">3.1</span> Number of cells per experiment</h3>
</section>
<div class="cell-output cell-output-stderr">
<pre><code>/projects/dan1/data/Brickman/conda/envs/scvi-1.0.0/lib/python3.10/site-packages/seaborn/categorical.py:82: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  plot_data = [np.asarray(s, float) for k, s in iter_data]</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="01_fetchngs_mouse_files/figure-html/cell-114-output-4.png"></p>
</div>
<section id="number-of-cells-cell-type" class="level3 cell-output cell-output-display" data-number="3.2">
<h3 data-number="3.2" data-anchor-id="number-of-cells-cell-type"><span class="header-section-number">3.2</span> Number of cells cell type</h3>
</section>
<div class="cell-output cell-output-display">
<p><img src="01_fetchngs_mouse_files/figure-html/cell-114-output-6.png"></p>
</div>
<section id="qc-plots" class="level3 cell-output cell-output-display" data-number="3.3">
<h3 data-number="3.3" data-anchor-id="qc-plots"><span class="header-section-number">3.3</span> QC plots</h3>
</section>
<div class="cell-output cell-output-display">
<p><img src="01_fetchngs_mouse_files/figure-html/cell-114-output-8.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="01_fetchngs_mouse_files/figure-html/cell-114-output-9.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="01_fetchngs_mouse_files/figure-html/cell-114-output-10.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="01_fetchngs_mouse_files/figure-html/cell-114-output-11.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="01_fetchngs_mouse_files/figure-html/cell-114-output-12.png"></p>
</div>
</div>
</section>
<section id="final-adjustments" class="level2" data-number="4">
<h2 data-number="4" data-anchor-id="final-adjustments"><span class="header-section-number">4</span> Final adjustments</h2>
<div class="cell" data-execution_count="114">
<div class="sourceCode cell-code" id="cb156"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb156-1"><a href="#cb156-1" aria-hidden="true" tabindex="-1"></a>adata</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="114">
<pre><code>AnnData object with n_obs × n_vars = 2022 × 54940
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig', 'stage', 'timepoint', 'n_genes_by_counts', 'total_counts', 'total_counts_mt', 'pct_counts_mt'
    var: 'gene_ids', 'gene_symbol', 'mt', 'n_cells_by_counts', 'mean_counts', 'pct_dropout_by_counts', 'total_counts'</code></pre>
</div>
</div>
<div class="cell" data-execution_count="115">
<div class="sourceCode cell-code" id="cb158"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb158-1"><a href="#cb158-1" aria-hidden="true" tabindex="-1"></a>sc.pl.violin(adata, [<span class="st">'n_genes_by_counts'</span>, <span class="st">'total_counts'</span>, <span class="st">'pct_counts_mt'</span>], jitter<span class="op">=</span><span class="fl">0.4</span>, multi_panel<span class="op">=</span><span class="va">True</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="01_fetchngs_mouse_files/figure-html/cell-116-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="116">
<div class="sourceCode cell-code" id="cb159"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb159-1"><a href="#cb159-1" aria-hidden="true" tabindex="-1"></a>sc.pp.filter_genes(adata, min_cells<span class="op">=</span><span class="dv">10</span>)</span>
<span id="cb159-2"><a href="#cb159-2" aria-hidden="true" tabindex="-1"></a>sc.pp.filter_cells(adata, max_genes<span class="op">=</span><span class="dv">20_000</span>)</span>
<span id="cb159-3"><a href="#cb159-3" aria-hidden="true" tabindex="-1"></a>sc.pp.filter_cells(adata, max_counts<span class="op">=</span><span class="dv">26_000_000</span>)</span>
<span id="cb159-4"><a href="#cb159-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb159-5"><a href="#cb159-5" aria-hidden="true" tabindex="-1"></a><span class="co"># adata = adata[adata.obs.pct_counts_mt &lt; 30]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="117">
<div class="sourceCode cell-code" id="cb160"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb160-1"><a href="#cb160-1" aria-hidden="true" tabindex="-1"></a>sc.pl.violin(adata, [<span class="st">'n_genes_by_counts'</span>, <span class="st">'total_counts'</span>, <span class="st">'pct_counts_mt'</span>], jitter<span class="op">=</span><span class="fl">0.4</span>, multi_panel<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb160-2"><a href="#cb160-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-3"><a href="#cb160-3" aria-hidden="true" tabindex="-1"></a>fig, ax <span class="op">=</span> plt.subplots(figsize<span class="op">=</span>(<span class="dv">20</span>, <span class="dv">5</span>), sharey<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb160-4"><a href="#cb160-4" aria-hidden="true" tabindex="-1"></a>sns.scatterplot(</span>
<span id="cb160-5"><a href="#cb160-5" aria-hidden="true" tabindex="-1"></a>    x<span class="op">=</span><span class="st">"total_counts"</span>, y<span class="op">=</span><span class="st">"n_genes_by_counts"</span>, ax<span class="op">=</span>ax, data<span class="op">=</span>adata.obs, hue<span class="op">=</span><span class="st">"ct"</span></span>
<span id="cb160-6"><a href="#cb160-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb160-7"><a href="#cb160-7" aria-hidden="true" tabindex="-1"></a><span class="co"># plt.axvline(x=26_000_000, c='r', linestyle='--')</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="01_fetchngs_mouse_files/figure-html/cell-118-output-1.png"></p>
</div>
<div class="cell-output cell-output-display" data-execution_count="117">
<pre><code>&lt;Axes: xlabel='total_counts', ylabel='n_genes_by_counts'&gt;</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="01_fetchngs_mouse_files/figure-html/cell-118-output-3.png"></p>
</div>
</div>
<div class="cell" data-execution_count="118">
<div class="sourceCode cell-code" id="cb162"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb162-1"><a href="#cb162-1" aria-hidden="true" tabindex="-1"></a>adata.layers[<span class="st">"counts"</span>] <span class="op">=</span> adata.X.copy()</span>
<span id="cb162-2"><a href="#cb162-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb162-3"><a href="#cb162-3" aria-hidden="true" tabindex="-1"></a>sc.pp.normalize_total(adata)</span>
<span id="cb162-4"><a href="#cb162-4" aria-hidden="true" tabindex="-1"></a>sc.pp.log1p(adata)</span>
<span id="cb162-5"><a href="#cb162-5" aria-hidden="true" tabindex="-1"></a>adata.raw <span class="op">=</span> adata</span>
<span id="cb162-6"><a href="#cb162-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb162-7"><a href="#cb162-7" aria-hidden="true" tabindex="-1"></a>adata</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="118">
<pre><code>AnnData object with n_obs × n_vars = 2004 × 34346
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig', 'stage', 'timepoint', 'n_genes_by_counts', 'total_counts', 'total_counts_mt', 'pct_counts_mt', 'n_genes', 'n_counts'
    var: 'gene_ids', 'gene_symbol', 'mt', 'n_cells_by_counts', 'mean_counts', 'pct_dropout_by_counts', 'total_counts', 'n_cells'
    uns: 'log1p'
    layers: 'counts'</code></pre>
</div>
</div>
<div class="cell" data-execution_count="119">
<div class="sourceCode cell-code" id="cb164"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb164-1"><a href="#cb164-1" aria-hidden="true" tabindex="-1"></a>adata.write(<span class="st">"../data/processed/01_mouse_reprocessed.h5ad"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="120">
<div class="sourceCode cell-code" id="cb165"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb165-1"><a href="#cb165-1" aria-hidden="true" tabindex="-1"></a>sc.tl.pca(adata, svd_solver<span class="op">=</span><span class="st">'arpack'</span>)</span>
<span id="cb165-2"><a href="#cb165-2" aria-hidden="true" tabindex="-1"></a>sc.pl.pca(adata, color<span class="op">=</span>[<span class="st">'ct'</span>, <span class="st">'batch'</span>, <span class="st">'technology'</span>, <span class="st">'experiment'</span>], frameon<span class="op">=</span><span class="va">False</span>, wspace<span class="op">=</span><span class="fl">0.4</span>, ncols<span class="op">=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>/projects/dan1/data/Brickman/conda/envs/scvi-1.0.0/lib/python3.10/site-packages/scanpy/plotting/_tools/scatterplots.py:392: UserWarning: No data for colormapping provided via 'c'. Parameters 'cmap' will be ignored
  cax = scatter(
/projects/dan1/data/Brickman/conda/envs/scvi-1.0.0/lib/python3.10/site-packages/scanpy/plotting/_tools/scatterplots.py:392: UserWarning: No data for colormapping provided via 'c'. Parameters 'cmap' will be ignored
  cax = scatter(
/projects/dan1/data/Brickman/conda/envs/scvi-1.0.0/lib/python3.10/site-packages/scanpy/plotting/_tools/scatterplots.py:392: UserWarning: No data for colormapping provided via 'c'. Parameters 'cmap' will be ignored
  cax = scatter(
/projects/dan1/data/Brickman/conda/envs/scvi-1.0.0/lib/python3.10/site-packages/scanpy/plotting/_tools/scatterplots.py:392: UserWarning: No data for colormapping provided via 'c'. Parameters 'cmap' will be ignored
  cax = scatter(</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="01_fetchngs_mouse_files/figure-html/cell-121-output-2.png"></p>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
});
</script>
</div> <!-- /content -->



</body></html>