<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.302">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Martin Proks">
<meta name="dcterms.date" content="2025-02-08">

<title>proks-salehin-et-al-2023 - 03 - mouse analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script>
<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../notebooks/00_figures_common.html">Reports</a></li><li class="breadcrumb-item"><a href="../notebooks/03_mouse_analysis.html">03 - mouse analysis</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">proks-salehin-et-al-2023</a> 
        <div class="sidebar-tools-main">
    <a href="https://twitter.com/LabBrickman" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-twitter"></i></a>
    <a href="https://github.com/brickmanlab" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../README.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">proks-salehin-et-al</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Reports</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/00_figures_common.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">00 - Figures (common)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/00_figures_human.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">00 - Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/00_figures_mouse.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">00 - Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/01_fetchngs_human.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">01 - NGS collection [human]</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/01_fetchngs_human-invitro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">01 - NGS collection [human]</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/01_fetchngs_mouse.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">01 - NGS collection [mouse]</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/02_human_integration.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">02 - human integration</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/02_human_tunning.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">02 - human tunning</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/02_mouse_integration.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">02 - mouse integration</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/02_mouse_tunning.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">02 - mouse tunning</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/03_human_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">03 - human analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/03_mouse_analysis.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">03 - mouse analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/04_human_cellrank.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Human CellRank</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/04_mouse_cellrank.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">04 - CellRank</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/05_human_classifier.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Re-annotation and classification</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/05_mouse_classifier.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">05 - mouse classifier</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/05_mouse_classifier_accuracy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">05 - mouse classifier accuracy</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/06_human_query_Rivron_SRR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Rivron blastoids cleaning</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/06_mouse_query.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">06 - mouse query</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/08_shap_explainers.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SHAP explainers for SCANVI</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/09_classification_explanation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">09 - Classification explanation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/10_mouse_beyond_celltypes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">05 - mouse classifier</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/11_human_query_Smith.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Austin Smith paper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/11_xgboost_vs_scanvi.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11 - XGBoost vs scANVI</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/12_upload_models.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Upload AI models</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#scvi" id="toc-scvi" class="nav-link active" data-scroll-target="#scvi"><span class="header-section-number">1</span> 1. SCVI</a>
  <ul class="collapse">
  <li><a href="#pca" id="toc-pca" class="nav-link" data-scroll-target="#pca"><span class="header-section-number">1.1</span> 1.0. PCA</a></li>
  <li><a href="#umap" id="toc-umap" class="nav-link" data-scroll-target="#umap"><span class="header-section-number">1.2</span> 1.1. UMAP</a></li>
  <li><a href="#phate" id="toc-phate" class="nav-link" data-scroll-target="#phate"><span class="header-section-number">1.3</span> 1.2. Phate</a></li>
  <li><a href="#t-sne" id="toc-t-sne" class="nav-link" data-scroll-target="#t-sne"><span class="header-section-number">1.4</span> 1.3. t-SNE</a></li>
  <li><a href="#force-directed-graph" id="toc-force-directed-graph" class="nav-link" data-scroll-target="#force-directed-graph"><span class="header-section-number">1.5</span> 1.4. Force Directed Graph</a></li>
  <li><a href="#dc" id="toc-dc" class="nav-link" data-scroll-target="#dc"><span class="header-section-number">1.6</span> 1.5. DC</a></li>
  </ul></li>
  <li><a href="#pseudotime" id="toc-pseudotime" class="nav-link" data-scroll-target="#pseudotime"><span class="header-section-number">2</span> 2. Pseudotime</a>
  <ul class="collapse">
  <li><a href="#dpt-package" id="toc-dpt-package" class="nav-link" data-scroll-target="#dpt-package"><span class="header-section-number">2.1</span> 2.1. DPT package</a></li>
  <li><a href="#scfates" id="toc-scfates" class="nav-link" data-scroll-target="#scfates"><span class="header-section-number">2.2</span> 2.2. scFATEs</a></li>
  <li><a href="#palantir" id="toc-palantir" class="nav-link" data-scroll-target="#palantir"><span class="header-section-number">2.3</span> 2.3. Palantir</a></li>
  </ul></li>
  <li><a href="#differentially-expressed-genes" id="toc-differentially-expressed-genes" class="nav-link" data-scroll-target="#differentially-expressed-genes"><span class="header-section-number">3</span> 3. Differentially expressed genes</a>
  <ul class="collapse">
  <li><a href="#markers" id="toc-markers" class="nav-link" data-scroll-target="#markers"><span class="header-section-number">3.1</span> 4. Markers</a></li>
  </ul></li>
  <li><a href="#save-model" id="toc-save-model" class="nav-link" data-scroll-target="#save-model"><span class="header-section-number">4</span> Save model</a></li>
  <li><a href="#other" id="toc-other" class="nav-link" data-scroll-target="#other"><span class="header-section-number">5</span> 4. OTHER</a>
  <ul class="collapse">
  <li><a href="#scgen" id="toc-scgen" class="nav-link" data-scroll-target="#scgen"><span class="header-section-number">5.1</span> 2. scGEN</a></li>
  <li><a href="#integration-stats" id="toc-integration-stats" class="nav-link" data-scroll-target="#integration-stats"><span class="header-section-number">5.2</span> 3. Integration stats</a></li>
  <li><a href="#prediction" id="toc-prediction" class="nav-link" data-scroll-target="#prediction"><span class="header-section-number">5.3</span> 3.1. Prediction</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">03 - mouse analysis</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Martin Proks </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 8, 2025</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<div class="cell" data-execution_count="1">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="op">!</span>which pip</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>/projects/dan1/data/Brickman/conda/envs/scvi-1.0.0/bin/pip</code></pre>
</div>
</div>
<div class="cell" data-execution_count="2">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>matplotlib inline</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> scvi</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> scanpy <span class="im">as</span> sc</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> scanpy.external <span class="im">as</span> sce</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> scFates <span class="im">as</span> scf</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> warnings</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> numba.core.errors <span class="im">import</span> NumbaDeprecationWarning</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>warnings.simplefilter(<span class="st">'ignore'</span>, category<span class="op">=</span>NumbaDeprecationWarning)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>warnings.simplefilter(<span class="st">'ignore'</span>, category<span class="op">=</span><span class="pp">FutureWarning</span>)</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>warnings.simplefilter(<span class="st">'ignore'</span>, category<span class="op">=</span><span class="pp">UserWarning</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>/projects/dan1/data/Brickman/conda/envs/scvi-1.0.0/lib/python3.10/site-packages/scvi/_settings.py:63: UserWarning: Since v1.0.0, scvi-tools no longer uses a random seed by default. Run `scvi.settings.seed = 0` to reproduce results from previous versions.
  self.seed = seed
/projects/dan1/data/Brickman/conda/envs/scvi-1.0.0/lib/python3.10/site-packages/scvi/_settings.py:70: UserWarning: Setting `dl_pin_memory_gpu_training` is deprecated in v1.0 and will be removed in v1.1. Please pass in `pin_memory` to the data loaders instead.
  self.dl_pin_memory_gpu_training = (
/projects/dan1/data/Brickman/conda/envs/scvi-1.0.0/lib/python3.10/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm</code></pre>
</div>
</div>
<div class="cell" data-execution_count="3">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>adata <span class="op">=</span> sc.read(<span class="st">"../results/02_mouse_integration/scvi/adata.h5ad"</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>adata</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="3">
<pre><code>AnnData object with n_obs × n_vars = 2004 × 3000
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig', 'stage', 'timepoint', 'n_genes_by_counts', 'total_counts', 'total_counts_mt', 'pct_counts_mt', 'n_genes', 'n_counts', 'batch_og', '_scvi_batch', '_scvi_labels'
    var: 'gene_ids', 'gene_symbol', 'mt', 'n_cells_by_counts', 'mean_counts', 'pct_dropout_by_counts', 'total_counts', 'n_cells', 'highly_variable', 'means', 'dispersions', 'dispersions_norm', 'highly_variable_nbatches', 'highly_variable_intersection'
    uns: '_scvi_manager_uuid', '_scvi_uuid', 'hvg', 'log1p'
    obsm: 'X_mde_scVI', 'X_scVI'
    layers: 'counts', 'scVI_normalized'</code></pre>
</div>
</div>
<section id="scvi" class="level2" data-number="1">
<h2 data-number="1" data-anchor-id="scvi"><span class="header-section-number">1</span> 1. SCVI</h2>
<div class="cell" data-execution_count="4">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>USE_REP <span class="op">=</span> <span class="st">'X_scVI'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="5">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Ideally find 15 clusters</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>sc.pp.neighbors(adata, use_rep<span class="op">=</span>USE_REP)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>sc.tl.leiden(adata, resolution<span class="op">=</span><span class="fl">0.8</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>2023-09-23 17:00:49.975107: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT</code></pre>
</div>
</div>
<div class="cell" data-execution_count="6">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>fig, ax <span class="op">=</span> plt.subplots(<span class="dv">1</span>, <span class="dv">2</span>, figsize<span class="op">=</span>[<span class="dv">25</span>, <span class="dv">5</span>])</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>adata.obs.groupby([<span class="st">'ct'</span>, <span class="st">'leiden'</span>]).size().unstack().plot(kind<span class="op">=</span><span class="st">'bar'</span>, stacked<span class="op">=</span><span class="va">True</span>, ax<span class="op">=</span>ax[<span class="dv">0</span>]).legend(loc<span class="op">=</span><span class="st">'right'</span>)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>adata.obs.groupby([<span class="st">'leiden'</span>, <span class="st">'ct'</span>]).size().unstack().plot(kind<span class="op">=</span><span class="st">'bar'</span>, stacked<span class="op">=</span><span class="va">True</span>, ax<span class="op">=</span>ax[<span class="dv">1</span>]).legend(loc<span class="op">=</span><span class="st">'right'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="6">
<pre><code>&lt;matplotlib.legend.Legend at 0x7f1a6a933fa0&gt;</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-7-output-2.png"></p>
</div>
</div>
<section id="pca" class="level3" data-number="1.1">
<h3 data-number="1.1" data-anchor-id="pca"><span class="header-section-number">1.1</span> 1.0. PCA</h3>
<div class="cell" data-execution_count="7">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>sc.tl.pca(adata)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>sc.pl.pca(adata, color<span class="op">=</span>[<span class="st">'stage'</span>, <span class="st">'ct'</span>], frameon<span class="op">=</span><span class="va">False</span>, ncols<span class="op">=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-8-output-1.png"></p>
</div>
</div>
</section>
<section id="umap" class="level3" data-number="1.2">
<h3 data-number="1.2" data-anchor-id="umap"><span class="header-section-number">1.2</span> 1.1. UMAP</h3>
<div class="cell" data-execution_count="8">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>sc.tl.umap(adata)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>sc.pl.umap(adata, color<span class="op">=</span>[<span class="st">'stage'</span>, <span class="st">'ct'</span>], frameon<span class="op">=</span><span class="va">False</span>, ncols<span class="op">=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-9-output-1.png"></p>
</div>
</div>
</section>
<section id="phate" class="level3" data-number="1.3">
<h3 data-number="1.3" data-anchor-id="phate"><span class="header-section-number">1.3</span> 1.2. Phate</h3>
<div class="cell" data-execution_count="9">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>sce.tl.phate(adata, n_jobs<span class="op">=</span><span class="dv">8</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>sce.pl.phate(adata, color<span class="op">=</span>[<span class="st">'stage'</span>, <span class="st">'ct'</span>], frameon<span class="op">=</span><span class="va">False</span>, ncols<span class="op">=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-10-output-1.png"></p>
</div>
</div>
</section>
<section id="t-sne" class="level3" data-number="1.4">
<h3 data-number="1.4" data-anchor-id="t-sne"><span class="header-section-number">1.4</span> 1.3. t-SNE</h3>
<div class="cell" data-execution_count="10">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>sc.tl.tsne(adata, n_jobs<span class="op">=</span><span class="dv">8</span>, perplexity<span class="op">=</span><span class="dv">300</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>sc.pl.tsne(adata, color<span class="op">=</span>[<span class="st">'stage'</span>, <span class="st">'ct'</span>], frameon<span class="op">=</span><span class="va">False</span>, ncols<span class="op">=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-11-output-1.png"></p>
</div>
</div>
</section>
<section id="force-directed-graph" class="level3" data-number="1.5">
<h3 data-number="1.5" data-anchor-id="force-directed-graph"><span class="header-section-number">1.5</span> 1.4. Force Directed Graph</h3>
<div class="cell" data-execution_count="11">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>sc.tl.draw_graph(adata, n_jobs<span class="op">=</span><span class="dv">8</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>sc.pl.draw_graph(adata, color<span class="op">=</span>[<span class="st">'experiment'</span>, <span class="st">'stage'</span>, <span class="st">'ct'</span>], frameon<span class="op">=</span><span class="va">False</span>, ncols<span class="op">=</span><span class="dv">3</span>, wspace<span class="op">=</span><span class="fl">0.4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-12-output-1.png"></p>
</div>
</div>
</section>
<section id="dc" class="level3" data-number="1.6">
<h3 data-number="1.6" data-anchor-id="dc"><span class="header-section-number">1.6</span> 1.5. DC</h3>
<div class="cell" data-execution_count="12">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Bug: https://github.com/scverse/scanpy/issues/2254</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>sc.tl.diffmap(adata)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>adata.obsm[<span class="st">"X_diffmap_"</span>] <span class="op">=</span> adata.obsm[<span class="st">"X_diffmap"</span>][:, <span class="dv">1</span>:] </span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>sc.pl.embedding(adata, <span class="st">"diffmap_"</span>, color<span class="op">=</span>[<span class="st">'stage'</span>, <span class="st">'ct'</span>], frameon<span class="op">=</span><span class="va">False</span>, ncols<span class="op">=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-13-output-1.png"></p>
</div>
</div>
</section>
</section>
<section id="pseudotime" class="level2" data-number="2">
<h2 data-number="2" data-anchor-id="pseudotime"><span class="header-section-number">2</span> 2. Pseudotime</h2>
<div class="cell" data-execution_count="13">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>adata</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="13">
<pre><code>AnnData object with n_obs × n_vars = 2004 × 3000
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig', 'stage', 'timepoint', 'n_genes_by_counts', 'total_counts', 'total_counts_mt', 'pct_counts_mt', 'n_genes', 'n_counts', 'batch_og', '_scvi_batch', '_scvi_labels', 'leiden'
    var: 'gene_ids', 'gene_symbol', 'mt', 'n_cells_by_counts', 'mean_counts', 'pct_dropout_by_counts', 'total_counts', 'n_cells', 'highly_variable', 'means', 'dispersions', 'dispersions_norm', 'highly_variable_nbatches', 'highly_variable_intersection'
    uns: '_scvi_manager_uuid', '_scvi_uuid', 'hvg', 'log1p', 'neighbors', 'leiden', 'pca', 'stage_colors', 'ct_colors', 'umap', 'tsne', 'draw_graph', 'experiment_colors', 'diffmap_evals'
    obsm: 'X_mde_scVI', 'X_scVI', 'X_pca', 'X_umap', 'X_phate', 'X_tsne', 'X_draw_graph_fa', 'X_diffmap', 'X_diffmap_'
    varm: 'PCs'
    layers: 'counts', 'scVI_normalized'
    obsp: 'distances', 'connectivities'</code></pre>
</div>
</div>
<section id="dpt-package" class="level3" data-number="2.1">
<h3 data-number="2.1" data-anchor-id="dpt-package"><span class="header-section-number">2.1</span> 2.1. DPT package</h3>
<div class="cell" data-execution_count="14">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>sc.pp.neighbors(adata, use_rep<span class="op">=</span>USE_REP)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>sc.tl.diffmap(adata)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>sc.tl.paga(adata, groups<span class="op">=</span><span class="st">'ct'</span>)</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>sc.pl.paga(adata, color<span class="op">=</span>[<span class="st">'ct'</span>], frameon<span class="op">=</span><span class="va">False</span>, fontoutline<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>sc.tl.draw_graph(adata, init_pos<span class="op">=</span><span class="st">'paga'</span>, n_jobs<span class="op">=</span><span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-15-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="17">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="cf">with</span> plt.rc_context({<span class="st">"figure.figsize"</span>: (<span class="dv">6</span>, <span class="dv">6</span>), <span class="st">"figure.dpi"</span>: (<span class="dv">300</span>)}):</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>    sc.pl.paga(adata, color<span class="op">=</span>[<span class="st">'ct'</span>], frameon<span class="op">=</span><span class="va">False</span>, fontoutline<span class="op">=</span><span class="va">True</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-16-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="18">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>adata.uns[<span class="st">'iroot'</span>] <span class="op">=</span> np.flatnonzero(adata.obs[<span class="st">'ct'</span>]  <span class="op">==</span> <span class="st">'Zygote'</span>)[<span class="dv">0</span>]</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>sc.tl.dpt(adata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="19">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>sc.pl.draw_graph(adata, color<span class="op">=</span>[<span class="st">'dpt_pseudotime'</span>, <span class="st">'ct'</span>], frameon<span class="op">=</span><span class="va">False</span>, ncols<span class="op">=</span><span class="dv">2</span>, cmap<span class="op">=</span><span class="st">'tab20'</span>)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>sc.pl.draw_graph(adata, color<span class="op">=</span>[<span class="st">'experiment'</span>, <span class="st">'ct'</span>], frameon<span class="op">=</span><span class="va">False</span>, ncols<span class="op">=</span><span class="dv">2</span>, cmap<span class="op">=</span><span class="st">'tab20'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-18-output-1.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-18-output-2.png"></p>
</div>
</div>
</section>
<section id="scfates" class="level3" data-number="2.2">
<h3 data-number="2.2" data-anchor-id="scfates"><span class="header-section-number">2.2</span> 2.2. scFATEs</h3>
<div class="cell" data-execution_count="20">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>sc.pp.neighbors(adata, use_rep<span class="op">=</span>USE_REP)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>sc.tl.draw_graph(adata, n_jobs<span class="op">=</span><span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="21">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>sc.pl.draw_graph(adata, color<span class="op">=</span>[<span class="st">'dpt_pseudotime'</span>, <span class="st">'ct'</span>], frameon<span class="op">=</span><span class="va">False</span>, ncols<span class="op">=</span><span class="dv">2</span>, cmap<span class="op">=</span><span class="st">'tab20'</span>)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>sc.pl.draw_graph(adata, color<span class="op">=</span>[<span class="st">'stage'</span>, <span class="st">'timepoint'</span>], frameon<span class="op">=</span><span class="va">False</span>, ncols<span class="op">=</span><span class="dv">2</span>, cmap<span class="op">=</span><span class="st">'tab20'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-20-output-1.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-20-output-2.png"></p>
</div>
</div>
<div class="cell" data-execution_count="27">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>sig <span class="op">=</span> scf.tl.explore_sigma(adata,</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>                         <span class="co"># Nodes=20,</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>                         Nodes<span class="op">=</span><span class="dv">60</span>,</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>                         use_rep<span class="op">=</span><span class="st">"X_draw_graph_fa"</span>,</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>                         sigmas<span class="op">=</span>[<span class="dv">1000</span>,<span class="dv">500</span>,<span class="dv">400</span>,<span class="dv">300</span>,<span class="dv">200</span>,<span class="dv">100</span>,<span class="dv">50</span>,<span class="dv">10</span>,<span class="dv">1</span>],</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>                         seed<span class="op">=</span><span class="dv">42</span>,plot<span class="op">=</span><span class="va">True</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-21-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="28">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>scf.tl.tree(adata,</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Nodes=30,</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>            Nodes<span class="op">=</span><span class="dv">60</span>,</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>            use_rep<span class="op">=</span><span class="st">"X_draw_graph_fa"</span>,</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>            method<span class="op">=</span><span class="st">"ppt"</span>,</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>            ppt_nsteps<span class="op">=</span><span class="dv">10</span>,</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>            ppt_sigma<span class="op">=</span>sig,</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>            ppt_lambda<span class="op">=</span><span class="dv">100</span>,</span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>            seed<span class="op">=</span><span class="dv">42</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>inferring a principal tree --&gt; parameters used 
    60 principal points, sigma = 500, lambda = 100, metric = euclidean
    fitting: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00&lt;00:00, 265.16it/s]
    not converged (error: 0.019733842722593525)
    finished (0:00:00) --&gt; added 
    .uns['ppt'], dictionnary containing inferred tree.
    .obsm['X_R'] soft assignment of cells to principal points.
    .uns['graph']['B'] adjacency matrix of the principal points.
    .uns['graph']['F'] coordinates of principal points in representation space.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="29">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>scf.pl.graph(adata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-23-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="35">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>scf.tl.root(adata, <span class="dv">58</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>node 58 selected as a root --&gt; added
    .uns['graph']['root'] selected root.
    .uns['graph']['pp_info'] for each PP, its distance vs root and segment assignment.
    .uns['graph']['pp_seg'] segments network information.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="36">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>scf.tl.pseudotime(adata,n_jobs<span class="op">=</span><span class="dv">10</span>,n_map<span class="op">=</span><span class="dv">10</span>,seed<span class="op">=</span><span class="dv">42</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>projecting cells onto the principal graph
    mappings: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:21&lt;00:00,  2.14s/it]
    finished (0:00:22) --&gt; added
    .obs['edge'] assigned edge.
    .obs['t'] pseudotime value.
    .obs['seg'] segment of the tree assigned.
    .obs['milestones'] milestone assigned.
    .uns['pseudotime_list'] list of cell projection from all mappings.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="37">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>scf.pl.trajectory(adata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-26-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="38">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>sc.pl.draw_graph(adata, color<span class="op">=</span>[<span class="st">'stage'</span>, <span class="st">'seg'</span>, <span class="st">'dpt_pseudotime'</span>, <span class="st">'t'</span>], frameon<span class="op">=</span><span class="va">False</span>, ncols<span class="op">=</span><span class="dv">2</span>, cmap<span class="op">=</span><span class="st">'tab20'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-27-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="39">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>scf.tl.dendrogram(adata, n_jobs<span class="op">=</span><span class="dv">8</span>)</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="cf">with</span> plt.rc_context({<span class="st">"figure.figsize"</span>: (<span class="dv">6</span>, <span class="dv">6</span>)}):</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>    scf.pl.dendrogram(adata,color<span class="op">=</span><span class="st">"seg"</span>)</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>    scf.pl.dendrogram(adata,color<span class="op">=</span><span class="st">"ct"</span>,legend_loc<span class="op">=</span><span class="st">"on data"</span>,color_milestones<span class="op">=</span><span class="va">True</span>,legend_fontoutline<span class="op">=</span><span class="va">True</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Generating dendrogram of tree
    segment : 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:02&lt;00:00,  2.18it/s]
    finished (0:00:02) --&gt; added 
    .obsm['X_dendro'], new embedding generated.
    .uns['dendro_segments'] tree segments used for plotting.</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-28-output-2.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-28-output-3.png"></p>
</div>
</div>
</section>
<section id="palantir" class="level3" data-number="2.3">
<h3 data-number="2.3" data-anchor-id="palantir"><span class="header-section-number">2.3</span> 2.3. Palantir</h3>
<p><strong>NOTE</strong>: doesn’t work, skipping</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># import logging</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="co"># logging.getLogger("matplotlib.font_manager").setLevel(logging.ERROR)</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a><span class="co"># sc.pp.neighbors(adata, use_rep="X_scANVI")</span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="co"># sce.tl.palantir(adata, use_adjacency_matrix=True, distances_key="distances")</span></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a><span class="co"># sc.tl.tsne(adata, n_pcs=2, use_rep='X_palantir_multiscale', perplexity=30)</span></span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a><span class="co"># sc.pl.embedding(adata, basis='X_palantir_multiscale', color=['ct', 'stage'], frameon=False, ncols=2)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="differentially-expressed-genes" class="level2" data-number="3">
<h2 data-number="3" data-anchor-id="differentially-expressed-genes"><span class="header-section-number">3</span> 3. Differentially expressed genes</h2>
<div class="cell" data-execution_count="40">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> filter_markers(df: pd.DataFrame, n_genes: <span class="bu">int</span> <span class="op">=</span> <span class="dv">5</span>, upper: <span class="bu">bool</span> <span class="op">=</span> <span class="va">False</span>):</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>    <span class="co"># significant only</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>    df <span class="op">=</span> df[</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>        (df[<span class="st">"is_de_fdr_0.05"</span>])</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>        <span class="op">&amp;</span> (df[<span class="st">"bayes_factor"</span>] <span class="op">&gt;</span> <span class="dv">3</span>)</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>        <span class="op">&amp;</span> (df[<span class="st">"non_zeros_proportion1"</span>] <span class="op">&gt;</span> <span class="fl">0.1</span>)</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>        <span class="op">&amp;</span> (df[<span class="st">"lfc_median"</span>] <span class="op">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>    ]</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>    comparisons <span class="op">=</span> df.comparison.unique()</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>    deg_df <span class="op">=</span> {}</span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> comparison <span class="kw">in</span> comparisons:</span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a>        cluster <span class="op">=</span> comparison.split(<span class="st">" "</span>)[<span class="dv">0</span>]</span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a>        markers <span class="op">=</span> (</span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a>            df.query(<span class="st">"comparison == @comparison"</span>)</span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a>            .sort_values(by<span class="op">=</span><span class="st">"lfc_median"</span>, ascending<span class="op">=</span><span class="va">False</span>)</span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a>            .head(n_genes)</span>
<span id="cb39-18"><a href="#cb39-18" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb39-19"><a href="#cb39-19" aria-hidden="true" tabindex="-1"></a>        deg_df[cluster] <span class="op">=</span> (</span>
<span id="cb39-20"><a href="#cb39-20" aria-hidden="true" tabindex="-1"></a>            markers.index.<span class="bu">str</span>.upper().tolist() <span class="cf">if</span> upper <span class="cf">else</span> markers.index.tolist()</span>
<span id="cb39-21"><a href="#cb39-21" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb39-22"><a href="#cb39-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-23"><a href="#cb39-23" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> deg_df</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="41">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>vae <span class="op">=</span> scvi.model.SCVI.load(<span class="st">"../results/02_mouse_integration/scvi/"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>INFO     File ../results/02_mouse_integration/scvi/model.pt already downloaded                                     </code></pre>
</div>
</div>
<div class="cell" data-execution_count="42">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>m_ct <span class="op">=</span> vae.differential_expression(groupby<span class="op">=</span><span class="st">"ct"</span>)</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>m_ct_filt <span class="op">=</span> filter_markers(m_ct, n_genes<span class="op">=</span><span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DE...: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 15/15 [00:16&lt;00:00,  1.11s/it]</code></pre>
</div>
</div>
<div class="cell" data-execution_count="43">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>pd.DataFrame.from_dict(m_ct_filt, orient<span class="op">=</span><span class="st">'index'</span>).transpose()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="43">
<div>


<table class="dataframe table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Zygote</th>
<th data-quarto-table-cell-role="th">2C</th>
<th data-quarto-table-cell-role="th">4C</th>
<th data-quarto-table-cell-role="th">8C</th>
<th data-quarto-table-cell-role="th">16C</th>
<th data-quarto-table-cell-role="th">E3.25-ICM</th>
<th data-quarto-table-cell-role="th">E3.25-TE</th>
<th data-quarto-table-cell-role="th">E3.5-ICM</th>
<th data-quarto-table-cell-role="th">E3.5-TE</th>
<th data-quarto-table-cell-role="th">E3.5-EPI</th>
<th data-quarto-table-cell-role="th">E3.5-PrE</th>
<th data-quarto-table-cell-role="th">E3.75-ICM</th>
<th data-quarto-table-cell-role="th">E4.5-TE</th>
<th data-quarto-table-cell-role="th">E4.5-EPI</th>
<th data-quarto-table-cell-role="th">E4.5-PrE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>aa545190</td>
<td>oog3</td>
<td>obox8</td>
<td>obox8</td>
<td>a430060f13rik</td>
<td>gm24289</td>
<td>gata3</td>
<td>1700021a07rik</td>
<td>tmprss2</td>
<td>fbxw10</td>
<td>gm19510</td>
<td>rnase12</td>
<td>s100a9</td>
<td>calca</td>
<td>has2</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>oosp3</td>
<td>au016765</td>
<td>gm4850</td>
<td>gm21731</td>
<td>gm11544</td>
<td>gm24920</td>
<td>gm24920</td>
<td>dglucy</td>
<td>adh1</td>
<td>nkain4</td>
<td>foxa2</td>
<td>iglv2</td>
<td>fasl</td>
<td>nphs1os</td>
<td>clic5</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>au016765</td>
<td>oosp3</td>
<td>b020004c17rik</td>
<td>gm11544</td>
<td>xkr9</td>
<td>gm24616</td>
<td>gm24289</td>
<td>gm807</td>
<td>slc15a2</td>
<td>lefty2</td>
<td>lefty2</td>
<td>nnmt</td>
<td>cald1</td>
<td>gm12690</td>
<td>spink1</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td>oog3</td>
<td>gm4850</td>
<td>oog3</td>
<td>a430060f13rik</td>
<td>dio1</td>
<td>xkr9</td>
<td>gm5907</td>
<td>aldh3a1</td>
<td>epn3</td>
<td>mal</td>
<td>sox17</td>
<td>c330004p14rik</td>
<td>rhox6</td>
<td>stmn2</td>
<td>clic6</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td>obox2</td>
<td>omt2a</td>
<td>gm36976</td>
<td>b020004c17rik</td>
<td>gm33508</td>
<td>gm25328</td>
<td>tmprss2</td>
<td>nkx6-2</td>
<td>cdx2</td>
<td>gm26564</td>
<td>dkk1</td>
<td>gm30717</td>
<td>h19</td>
<td>col18a1</td>
<td>habp2</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5</td>
<td>c87499</td>
<td>aa545190</td>
<td>omt2a</td>
<td>af067063</td>
<td>gm24289</td>
<td>snord35a</td>
<td>cryab</td>
<td>crygd</td>
<td>gata3</td>
<td>cobl</td>
<td>bmper</td>
<td>gm6284</td>
<td>elf5</td>
<td>fgf15</td>
<td>tfec</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">6</td>
<td>e330034g19rik</td>
<td>tcl1b5</td>
<td>d7ertd443e</td>
<td>d7ertd443e</td>
<td>gm5435</td>
<td>cck</td>
<td>gm6334</td>
<td>sox2</td>
<td>dppa1</td>
<td>cd3g</td>
<td>lmo2</td>
<td>gpx8</td>
<td>hpgd</td>
<td>gm26564</td>
<td>rhox6</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">7</td>
<td>gm43269</td>
<td>tcl1b4</td>
<td>gm21731</td>
<td>gm13339</td>
<td>gm12446</td>
<td>gm25636</td>
<td>au021092</td>
<td>mal</td>
<td>id2</td>
<td>myh13</td>
<td>f2r</td>
<td>gm13339</td>
<td>ankrd1</td>
<td>ctsw</td>
<td>ttr</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">8</td>
<td>tcl1b4</td>
<td>c87499</td>
<td>gdap1</td>
<td>dynap</td>
<td>dynap</td>
<td>dio1</td>
<td>arg2</td>
<td>lmo2</td>
<td>cryab</td>
<td>gm12688</td>
<td>serpinh1</td>
<td>p3h2</td>
<td>slc16a3</td>
<td>prss50</td>
<td>chst15</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">9</td>
<td>oosp2</td>
<td>oosp2</td>
<td>spz1</td>
<td>xkr9</td>
<td>snord35a</td>
<td>gm33508</td>
<td>nat8</td>
<td>gm5662</td>
<td>serpinb6b</td>
<td>gm807</td>
<td>flrt3</td>
<td>stfa1</td>
<td>tmprss2</td>
<td>c330004p14rik</td>
<td>p3h2</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div class="cell" data-execution_count="44">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>sc.pl.dotplot(adata, m_ct_filt, groupby<span class="op">=</span><span class="st">'ct'</span>, dendrogram<span class="op">=</span><span class="va">False</span>, standard_scale<span class="op">=</span><span class="st">'var'</span>)</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>sc.pl.matrixplot(adata, m_ct_filt, groupby<span class="op">=</span><span class="st">'ct'</span>, dendrogram<span class="op">=</span><span class="va">False</span>, standard_scale<span class="op">=</span><span class="st">'var'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-34-output-1.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-34-output-2.png"></p>
</div>
</div>
<section id="markers" class="level3" data-number="3.1">
<h3 data-number="3.1" data-anchor-id="markers"><span class="header-section-number">3.1</span> 4. Markers</h3>
<ul>
<li>zygote/2C: zscan4a, zscan4b, zscan4c, zscan4d, zscan4f</li>
<li>8c: carm1, cdx2, pard3, pard6, apkc</li>
<li>totipotency: cdx2, gata6, nanog</li>
<li>apical domain: apkc, par2, par6, ezrin</li>
<li>basolateral domain: cdh1, lgl, par1</li>
<li>TE: cdx2, nanog, gata6</li>
<li>inner cell mass: pou5f1, nanog, gata6, sox2, lats, klf4, dppa3, esrrb</li>
<li>primitive endoderm: pou5f1, gata6, gata4, dab2, sox7, sox17, pdgfra, hnf4a</li>
<li>epiblast: pou5f1, nanog, sox2, fgf4, pecam, fgf5, bmp4, klf2</li>
</ul>
<hr>
<ul>
<li>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1994725/</li>
<li>https://link.springer.com/article/10.1007/s00441-021-03530-8</li>
<li>https://journals.biologists.com/dev/article/143/7/1063/47817/Lineage-specification-in-the-mouse-preimplantation</li>
</ul>
<div class="cell" data-execution_count="45">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>sc.tl.embedding_density(adata, basis<span class="op">=</span><span class="st">'draw_graph_fa'</span>, groupby<span class="op">=</span><span class="st">'ct'</span>)</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>sc.pl.embedding_density(adata, basis<span class="op">=</span><span class="st">'draw_graph_fa'</span>, key<span class="op">=</span><span class="st">'draw_graph_fa_density_ct'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-35-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="46">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>lineage_markers <span class="op">=</span> pd.read_excel(<span class="st">"../data/external/mouse_lineage_markers.xlsx"</span>, sheet_name<span class="op">=</span><span class="st">"Sheet1"</span>).fillna(<span class="st">''</span>)</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>lineage_markers</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="46">
<div>


<table class="dataframe table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">Zygote</th>
<th data-quarto-table-cell-role="th">2C</th>
<th data-quarto-table-cell-role="th">4C</th>
<th data-quarto-table-cell-role="th">8C</th>
<th data-quarto-table-cell-role="th">16C</th>
<th data-quarto-table-cell-role="th">TE</th>
<th data-quarto-table-cell-role="th">ICM</th>
<th data-quarto-table-cell-role="th">PrE</th>
<th data-quarto-table-cell-role="th">EPI</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">0</td>
<td>Zswim3</td>
<td>Zscan4a</td>
<td>Sin3a</td>
<td>Pou5f1</td>
<td>Pou5f1</td>
<td>Cdx2</td>
<td>Pou5f1</td>
<td>Gata6</td>
<td>Pou5f1</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">1</td>
<td>Trim28</td>
<td>Zscan4b</td>
<td>Clock</td>
<td>Nanog</td>
<td>Nanog</td>
<td>Gata3</td>
<td>Gata6</td>
<td>Gata4</td>
<td>Nanog</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">2</td>
<td>Padi6</td>
<td>Zscan4c</td>
<td>Sox21</td>
<td>Sox2</td>
<td>Sox2</td>
<td>Gata2</td>
<td>Nanog</td>
<td>Sox7</td>
<td>Sox2</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">3</td>
<td></td>
<td>Zscan4d</td>
<td>Carm1</td>
<td>Cdx2</td>
<td>Cdx2</td>
<td>Eomes</td>
<td>Tfcp2l1</td>
<td>Sox17</td>
<td>Fgf4</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">4</td>
<td></td>
<td>Zscan4f</td>
<td></td>
<td>Yap1</td>
<td>Esrrb</td>
<td>Lats2</td>
<td>Dppa3</td>
<td>Pdgfra</td>
<td>Klf2</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">5</td>
<td></td>
<td>Zfp352</td>
<td></td>
<td>Eifg1</td>
<td>Prdm14</td>
<td>Tfap2c</td>
<td>Esrrb</td>
<td>Dab2</td>
<td></td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">6</td>
<td></td>
<td>Usp17ld</td>
<td></td>
<td>Prdm14</td>
<td>Tfap2c</td>
<td></td>
<td>Prdm14</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">7</td>
<td></td>
<td>Rxra</td>
<td></td>
<td></td>
<td>Gata3</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">8</td>
<td></td>
<td>Tmem92</td>
<td></td>
<td></td>
<td>Gata2</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">9</td>
<td></td>
<td>Pramef25</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">10</td>
<td></td>
<td>Dux</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div class="cell" data-execution_count="47">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> stage, genes <span class="kw">in</span> lineage_markers.to_dict(<span class="st">'list'</span>).items():</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>    common_genes <span class="op">=</span> adata.var_names.intersection([<span class="bu">str</span>.lower(g) <span class="cf">for</span> g <span class="kw">in</span> genes <span class="cf">if</span> g <span class="op">!=</span> <span class="st">''</span>])</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>    <span class="co"># print(stage, common_genes)</span></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>    sc.pl.draw_graph(adata, color<span class="op">=</span>[<span class="st">'stage'</span>, <span class="op">*</span>common_genes], title<span class="op">=</span>[stage, <span class="op">*</span>common_genes],</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>                     groups<span class="op">=</span>stage, legend_loc<span class="op">=</span><span class="va">None</span>, frameon<span class="op">=</span><span class="va">False</span>, ncols<span class="op">=</span><span class="bu">len</span>(common_genes) <span class="op">+</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-37-output-1.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-37-output-2.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-37-output-3.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-37-output-4.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-37-output-5.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-37-output-6.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-37-output-7.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-37-output-8.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="03_mouse_analysis_files/figure-html/cell-37-output-9.png"></p>
</div>
</div>
</section>
</section>
<section id="save-model" class="level2" data-number="4">
<h2 data-number="4" data-anchor-id="save-model"><span class="header-section-number">4</span> Save model</h2>
<div class="cell" data-execution_count="48">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>adata.write(<span class="st">"../results/03_mouse.processed.h5ad"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="other" class="level2" data-number="5">
<h2 data-number="5" data-anchor-id="other"><span class="header-section-number">5</span> 4. OTHER</h2>
<section id="scgen" class="level3" data-number="5.1">
<h3 data-number="5.1" data-anchor-id="scgen"><span class="header-section-number">5.1</span> 2. scGEN</h3>
<p><code>scGen</code> returns imputed counts only. In this case we are not able to run DEGs because we need to have normalized counts. The <code>predict</code> function also won’t work because we don’t have any perturbation in the dataset.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> scgen</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>mscgen <span class="op">=</span> scgen.SCGEN.load(<span class="st">"../results/02_mouse_integration/scgen/"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>corrected_adata <span class="op">=</span> mscgen.batch_removal()</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>corrected_adata</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>sc.pp.neighbors(corrected_adata, use_rep<span class="op">=</span><span class="st">'corrected_latent'</span>)</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>sc.tl.draw_graph(corrected_adata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>sc.pl.draw_graph(corrected_adata, color<span class="op">=</span>[<span class="st">'ct'</span>, <span class="st">'Stage'</span>], wspace<span class="op">=</span><span class="fl">0.4</span>, frameon<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>sc.tl.umap(corrected_adata)</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>sc.pl.umap(corrected_adata, color<span class="op">=</span>[<span class="st">'batch'</span>, <span class="st">'Stage'</span>], wspace<span class="op">=</span><span class="fl">0.4</span>, frameon<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="integration-stats" class="level3" data-number="5.2">
<h3 data-number="5.2" data-anchor-id="integration-stats"><span class="header-section-number">5.2</span> 3. Integration stats</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> rich <span class="im">import</span> <span class="bu">print</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> scib_metrics.benchmark <span class="im">import</span> Benchmarker</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>bm <span class="op">=</span> Benchmarker(</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>    adata,</span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>    batch_key<span class="op">=</span><span class="st">"batch"</span>,</span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>    label_key<span class="op">=</span><span class="st">"ct"</span>,</span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a>    embedding_obsm_keys<span class="op">=</span>[<span class="st">"X_pca"</span>, <span class="st">"X_scVI"</span>, <span class="st">"X_scANVI"</span>, <span class="st">"X_scgen"</span>],</span>
<span id="cb57-6"><a href="#cb57-6" aria-hidden="true" tabindex="-1"></a>    n_jobs<span class="op">=-</span><span class="dv">1</span>,</span>
<span id="cb57-7"><a href="#cb57-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb57-8"><a href="#cb57-8" aria-hidden="true" tabindex="-1"></a>bm.benchmark()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>bm.plot_results_table(min_max_scale<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> pysankey <span class="im">import</span> sankey</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>run ..<span class="op">/</span>scripts<span class="op">/</span>helpers.py</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>sc.tl.dendrogram(adata, groupby<span class="op">=</span><span class="st">"leiden"</span>, key_added<span class="op">=</span><span class="st">"dendrogram_leiden"</span>)</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>sc.tl.dendrogram(adata, groupby<span class="op">=</span><span class="st">"ct"</span>, key_added<span class="op">=</span><span class="st">"dendrogram_ct"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="cf">with</span> plt.rc_context({<span class="st">"figure.figsize"</span>: [<span class="dv">20</span>, <span class="dv">20</span>]}):</span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>    sc.pl.correlation_matrix(adata, groupby<span class="op">=</span><span class="st">"ct"</span>, show_correlation_numbers<span class="op">=</span><span class="va">True</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>_ <span class="op">=</span> sc.pl.dendrogram(adata, groupby<span class="op">=</span><span class="st">"ct"</span>, orientation<span class="op">=</span><span class="st">'top'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="prediction" class="level3" data-number="5.3">
<h3 data-number="5.3" data-anchor-id="prediction"><span class="header-section-number">5.3</span> 3.1. Prediction</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> get_predictions(lvae: scvi.model.SCANVI, threshold: <span class="bu">float</span> <span class="op">=</span> <span class="fl">0.85</span>):</span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>    df <span class="op">=</span> pd.DataFrame(<span class="bu">zip</span>(predictions.idxmax(axis<span class="op">=</span><span class="dv">1</span>), predictions.<span class="bu">max</span>(axis<span class="op">=</span><span class="dv">1</span>)), columns<span class="op">=</span>[<span class="st">'predicted'</span>, <span class="st">'score'</span>])</span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>    df[<span class="st">'predicted_filt'</span>] <span class="op">=</span> <span class="st">'Unknown'</span></span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>    df.loc[df.score <span class="op">&gt;=</span> threshold, <span class="st">'predicted_filt'</span>] <span class="op">=</span> df.loc[df.score <span class="op">&gt;=</span> threshold, <span class="st">'predicted'</span>]</span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>    df[<span class="st">'uncertainty'</span>] <span class="op">=</span> np.maximum(<span class="dv">1</span> <span class="op">-</span> df.score, <span class="dv">0</span>)</span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> df</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>lvae <span class="op">=</span> scvi.model.SCANVI.load(<span class="st">"../results/02-mouse/"</span>)</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>adata.obs[[<span class="st">"predicted"</span>, <span class="st">"score"</span>, <span class="st">"predicted_filt"</span>, <span class="st">"uncertainty"</span>]] <span class="op">=</span> get_predictions(lvae).values</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>sc.pl.draw_graph(adata, color<span class="op">=</span>[<span class="st">'ct'</span>, <span class="st">'predicted'</span>, <span class="st">'uncertainty'</span>], frameon<span class="op">=</span><span class="va">False</span>, ncols<span class="op">=</span><span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>df <span class="op">=</span> adata.obs.groupby([<span class="st">"ct"</span>, <span class="st">"predicted"</span>]).size().unstack(fill_value<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>norm_df <span class="op">=</span> df <span class="op">/</span> df.<span class="bu">sum</span>(axis<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>plt.figure(figsize<span class="op">=</span>(<span class="dv">8</span>, <span class="dv">8</span>))</span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>_ <span class="op">=</span> plt.pcolor(norm_df)</span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a>_ <span class="op">=</span> plt.xticks(np.arange(<span class="fl">0.5</span>, <span class="bu">len</span>(df.columns), <span class="dv">1</span>), df.columns, rotation<span class="op">=</span><span class="dv">90</span>)</span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a>_ <span class="op">=</span> plt.yticks(np.arange(<span class="fl">0.5</span>, <span class="bu">len</span>(df.index), <span class="dv">1</span>), df.index)</span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a>_ <span class="op">=</span> plt.colorbar()</span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a>plt.xlabel(<span class="st">"Predicted"</span>)</span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a>plt.ylabel(<span class="st">"Observed"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Stats</span></span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> sklearn.metrics <span class="im">import</span> balanced_accuracy_score</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a>balanced_accuracy_score(adata.obs[<span class="st">"ct"</span>].values, adata.obs[<span class="st">"predicted"</span>].values)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
});
</script>
</div> <!-- /content -->



</body></html>